// composio-core/lib/src/cli/add.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const chalk_1 = __importDefault(require("chalk"));
const sdk_1 = require("../sdk");
const inquirer_1 = __importDefault(require("inquirer"));
const open_1 = __importDefault(require("open"));
class AddCommand {
    constructor(program) {
        this.program = program;
        this.program
            .command("add")
            .description("Add a new app")
            .argument("<app-name>", "The name of the app")
            .option("-f, --force", "Force the connection setup")
            .action(this.handleAction.bind(this));
    }
    async handleAction(appName, options) {
        const composioClient = new sdk_1.Composio();
        let integration = await composioClient.integrations.list({
            // @ts-ignore
            appName: appName.toLowerCase(),
        });
        if ((integration === null || integration === void 0 ? void 0 : integration.items.length) === 0) {
            integration = (await this.createIntegration(appName));
        }
        const firstIntegration = integration === null || integration === void 0 ? void 0 : integration.items[0];
        if (!firstIntegration) {
            console.log(chalk_1.default.red("No integration found or created"));
            return;
        }
        const connection = await composioClient.connectedAccounts.list({
            // @ts-ignore
            integrationId: firstIntegration.id,
        });
        if (connection.items.length > 0 && !options.force) {
            await this.shouldForceConnectionSetup();
        }
        // @ts-ignore
        await this.setupConnections(firstIntegration.id);
    }
    async shouldForceConnectionSetup() {
        const { shouldForce } = await inquirer_1.default.prompt([
            {
                type: 'confirm',
                name: 'shouldForce',
                message: 'A connection already exists. Do you want to force a new connection?',
                default: false,
            },
        ]);
        if (!shouldForce) {
            console.log(chalk_1.default.yellow('Operation cancelled. Existing connection will be used.'));
            process.exit(0);
        }
    }
    async waitUntilConnected(connectedAccountId, timeout = 30000) {
        const composioClient = new sdk_1.Composio();
        const startTime = Date.now();
        const pollInterval = 3000; // 3 seconds
        while (Date.now() - startTime < timeout) {
            try {
                const data = (await composioClient.connectedAccounts.get({
                    connectedAccountId: connectedAccountId,
                }));
                if (data.status === "ACTIVE") {
                    return;
                }
            }
            catch (error) {
                console.error("Error checking connection status:", error);
            }
            await new Promise((resolve) => setTimeout(resolve, pollInterval));
        }
        throw new Error(`Connection did not become active within ${timeout / 1000} seconds`);
    }
    async setupConnections(integrationId) {
        const composioClient = new sdk_1.Composio();
        const data = await composioClient.integrations.get({ integrationId });
        const { expectedInputFields } = data;
        const config = await this.collectInputFields(expectedInputFields, true);
        const connectionData = await composioClient.connectedAccounts.create({
            integrationId,
            data: config,
        });
        if (connectionData.connectionStatus === "ACTIVE") {
            console.log(chalk_1.default.green("Connection created successfully"));
        }
        if (connectionData.redirectUrl) {
            console.log(chalk_1.default.white("Redirecting to the app"), chalk_1.default.blue(connectionData.redirectUrl));
            (0, open_1.default)(connectionData.redirectUrl);
            await this.waitUntilConnected(connectionData.connectedAccountId);
            console.log(chalk_1.default.green("Connection is active"));
            process.exit(0);
        }
    }
    async createIntegration(appName) {
        var _a, _b;
        const composioClient = new sdk_1.Composio();
        const app = await composioClient.apps.get({
            appKey: appName.toLowerCase(),
        });
        if (app.no_auth) {
            console.log(chalk_1.default.green(`The app '${appName}' does not require authentication. You can connect it directly.\n`));
            process.exit(0);
        }
        const config = {};
        const { integrationName } = await inquirer_1.default.prompt({
            type: "input",
            name: "integrationName",
            message: "Enter the app name",
        });
        if (!integrationName) {
            console.log(chalk_1.default.red("Integration name is required"));
            return null;
        }
        config.name = integrationName;
        // @ts-ignore
        const authSchema = (_a = app.auth_schemes[0]) === null || _a === void 0 ? void 0 : _a.auth_mode;
        // @ts-ignore
        if (((_b = app === null || app === void 0 ? void 0 : app.testConnectors) === null || _b === void 0 ? void 0 : _b.length) > 0 || app.no_auth) {
            let useComposioAuth = false;
            if (!app.no_auth) {
                const { doYouWantToUseComposioAuth } = await inquirer_1.default.prompt({
                    type: "confirm",
                    name: "doYouWantToUseComposioAuth",
                    message: "Do you want to use Composio Auth?",
                });
                useComposioAuth = doYouWantToUseComposioAuth;
            }
            config.useComposioAuth = useComposioAuth;
            return this.setupIntegration(app, authSchema, useComposioAuth, config);
        }
        const authConfig = await this.collectInputFields(
        // @ts-ignore
        app.auth_schemes[0].fields);
        return this.setupIntegration(app, authSchema, false, authConfig);
    }
    async collectInputFields(fields, isConnection = false) {
        const config = {};
        for (const field of fields) {
            if (field.expected_from_customer && !isConnection) {
                continue;
            }
            const { [field.name]: value } = await inquirer_1.default.prompt({
                type: "input",
                name: field.name,
                message: field.displayName || field.display_name,
            });
            if (value) {
                config[field.name] = value;
            }
        }
        return config;
    }
    async setupIntegration(app, authMode, useComposioAuth, config) {
        const composioClient = new sdk_1.Composio();
        await composioClient.integrations.create({
            appId: app.id,
            authScheme: authMode,
            useComposioAuth,
            authConfig: config,
        });
        return composioClient.integrations.list({
            // @ts-ignore
            appName: app.name.toLowerCase(),
        });
    }
}
exports.default = AddCommand;
