// composio-core/lib/src/cli/connections.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionsGetCommand = void 0;
const chalk_1 = __importDefault(require("chalk"));
const client_1 = __importDefault(require("../sdk/client/client"));
const config_1 = require("../sdk/utils/config");
class ConnectionsCommand {
    constructor(program) {
        this.program = program;
        const command = this.program.command("connections");
        command
            .description("List all connections you have access to")
            .option("-a, --active", "Show only active connections")
            .action(this.handleAction.bind(this));
        new ConnectionsGetCommand(command);
    }
    async handleAction(options) {
        (0, config_1.getAPISDK)();
        const { data, error } = await client_1.default.connections.getConnections({
            query: options.active ? { status: "ACTIVE" } : {},
            throwOnError: false,
        });
        if (error) {
            console.log(chalk_1.default.red(error.message));
            return;
        }
        for (const connection of (data === null || data === void 0 ? void 0 : data.items) || []) {
            console.log(chalk_1.default.cyan(`• ${chalk_1.default.bold("Id")}: ${connection.id}`));
            console.log(chalk_1.default.magenta(`  ${chalk_1.default.bold("App")}: ${connection.appName}`));
            console.log(chalk_1.default.yellow(`  ${chalk_1.default.bold("Status")}: ${connection.status}`));
            console.log(""); // Add an empty line for better readability between connections
        }
    }
}
exports.default = ConnectionsCommand;
class ConnectionsGetCommand {
    constructor(program) {
        this.program = program;
        this.program
            .command("get")
            .description("Get a connection by id")
            .argument("<id>", "Connection id (required)")
            .action(this.handleAction.bind(this));
    }
    async handleAction(id) {
        (0, config_1.getAPISDK)();
        const { data, error } = await client_1.default.connections.getConnection({
            path: { connectedAccountId: id },
            throwOnError: false,
        });
        if (error) {
            console.log(chalk_1.default.red(error.message));
            return;
        }
        for (const [key, value] of Object.entries(data)) {
            console.log(`- ${chalk_1.default.cyan.bold(key)}: ${JSON.stringify(value, null, 2)}`);
        }
    }
}
exports.ConnectionsGetCommand = ConnectionsGetCommand;
