// composio-core/lib/src/cli/login.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const chalk_1 = __importDefault(require("chalk"));
const inquirer_1 = __importDefault(require("inquirer"));
const open_1 = __importDefault(require("open"));
const config_1 = require("../sdk/utils/config");
const client_1 = __importDefault(require("../sdk/client/client"));
const constants_1 = require("./src/constants");
class LoginCommand {
    constructor(program) {
        this.program = program;
        this.program
            .command("login")
            .option("-n, --no-browser", "No browser will be opened, you will have to manually copy the link and paste it in your browser")
            .description("Authenticate and login to Composio")
            .action(this.handleAction.bind(this));
    }
    async handleAction(options) {
        (0, config_1.getAPISDK)();
        const { apiKey, baseURL } = (0, config_1.getClientBaseConfig)();
        if (apiKey) {
            console.log(chalk_1.default.yellow("✨ You are already authenticated and ready to use Composio! ✨\n"));
            return;
        }
        const { data, error } = await client_1.default.cli.handleCliCodeExchange({});
        if (!!error) {
            console.log(chalk_1.default.red(error.message));
            return;
        }
        const cliKey = data === null || data === void 0 ? void 0 : data.key;
        const loginUrl = `${constants_1.FRONTEND_BASE_URL}?cliKey=${cliKey}`;
        this.displayLoginInstructions(loginUrl, options.browser);
        const authCode = await this.promptForAuthCode();
        await this.verifyAndSetupCli(cliKey, authCode, baseURL);
    }
    displayLoginInstructions(url, openBrowser) {
        if (openBrowser) {
            (0, open_1.default)(url);
        }
        console.log("> Please login using the following link");
        console.log(chalk_1.default.cyan(url));
    }
    async promptForAuthCode() {
        const { authCode } = await inquirer_1.default.prompt({
            type: "input",
            name: "authCode",
            message: "Enter authentication code:",
        });
        return authCode;
    }
    async verifyAndSetupCli(cliKey, authCode, baseURL) {
        const { data, error } = await client_1.default.cli.handleCliCodeVerification({
            query: { key: cliKey, code: authCode },
            throwOnError: false,
        });
        if (error) {
            throw new Error(error.message);
        }
        const apiKeyFromServer = data === null || data === void 0 ? void 0 : data.apiKey;
        (0, config_1.setCliConfig)(apiKeyFromServer, "");
        console.log(chalk_1.default.yellow("✨ You are authenticated and ready to use Composio! ✨\n"));
    }
}
exports.default = LoginCommand;
