// composio-core/lib/src/cli/triggers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActiveTriggers = exports.TriggerDisable = exports.TriggerAdd = void 0;
const chalk_1 = __importDefault(require("chalk"));
const config_1 = require("../sdk/utils/config");
const sdk_1 = require("../sdk");
const inquirer_1 = __importDefault(require("inquirer"));
class ConnectionsCommand {
    constructor(program) {
        this.program = program;
        const command = this.program.command("triggers");
        command
            .description("Manage and list triggers")
            .option("--id <text>", "Filter by trigger id")
            .option("--app <text>", "Filter by app name")
            .action(this.handleAction.bind(this));
        new TriggerAdd(command);
        new TriggerDisable(command);
        new ActiveTriggers(command);
    }
    async handleAction(options) {
        const { active, id, app } = options;
        const client = (0, config_1.getAPISDK)();
        const { data, error } = await client.triggers.listTriggers({
            query: {
                ...(!!active && { showEnabledOnly: true }),
                ...(!!app && { appNames: app }),
                ...(!!id && { triggerIds: id }),
            },
            throwOnError: false,
        });
        if (error) {
            console.log(chalk_1.default.red(error.message));
            return;
        }
        for (const trigger of data || []) {
            const typedTrigger = trigger;
            console.log(chalk_1.default.cyan(`  ${chalk_1.default.bold("Name")}:`), chalk_1.default.white(typedTrigger.appName));
            console.log(chalk_1.default.cyan(`  ${chalk_1.default.bold("Enum")}:`), chalk_1.default.white(typedTrigger.enum));
            console.log(chalk_1.default.cyan(`  ${chalk_1.default.bold("Description")}:`), chalk_1.default.white(typedTrigger.description));
            console.log(""); // Add an empty line for better readability between triggers
        }
    }
}
exports.default = ConnectionsCommand;
class TriggerAdd {
    constructor(program) {
        this.program = program;
        this.program
            .command("add")
            .description("Add a new trigger")
            .argument("<trigger>", "The trigger name")
            .action(this.handleAction.bind(this));
    }
    async handleAction(triggerName) {
        const composioClient = new sdk_1.Composio();
        const data = (await composioClient.triggers.list()).find(
        // @ts-ignore
        (trigger) => trigger.enum.toLowerCase() === triggerName.toLowerCase());
        if (!data) {
            console.log(chalk_1.default.red(`Trigger ${triggerName} not found`));
            return;
        }
        const appName = data.appName;
        const connection = await composioClient
            .getEntity("default")
            .getConnection(appName);
        if (!connection) {
            console.log(chalk_1.default.red(`Connection to app ${appName} not found`));
            console.log(`Connect to the app by running: ${chalk_1.default.cyan(`composio add ${appName}`)}`);
            return;
        }
        const properties = data.config.properties;
        const requiredProperties = data.config.required;
        const configValue = {};
        for (const key in properties) {
            if (requiredProperties.includes(key)) {
                const answer = await inquirer_1.default.prompt([
                    {
                        type: "input",
                        name: key,
                        message: `Enter the value for ${key}`,
                    },
                ]);
                configValue[key] = answer[key];
            }
        }
        const triggerSetupData = await composioClient.triggers.setup(connection.id, triggerName, configValue);
        console.log(chalk_1.default.green(`Trigger ${triggerName} setup to app ${appName} with id ${triggerSetupData === null || triggerSetupData === void 0 ? void 0 : triggerSetupData.triggerId}`));
    }
}
exports.TriggerAdd = TriggerAdd;
class TriggerDisable {
    constructor(program) {
        this.program = program;
        this.program
            .command("disable")
            .description("Disable an existing trigger")
            .argument("<triggerid>", "The trigger id")
            .action(this.handleAction.bind(this));
    }
    async handleAction(triggerId) {
        const composioClient = new sdk_1.Composio();
        try {
            await composioClient.triggers.disable({ triggerId });
            console.log(chalk_1.default.green(`Trigger ${triggerId} disabled`));
        }
        catch (error) {
            console.log(chalk_1.default.red(`Error disabling trigger ${triggerId}: ${error}`));
        }
    }
}
exports.TriggerDisable = TriggerDisable;
class ActiveTriggers {
    constructor(program) {
        this.program = program;
        this.program
            .command("active")
            .description("Show list of currently active triggers")
            .action(this.handleAction.bind(this));
    }
    async handleAction() {
        const composioClient = new sdk_1.Composio();
        const triggers = await composioClient.activeTriggers.list();
        for (const trigger of triggers) {
            console.log(`Id: ${chalk_1.default.bold(trigger.id)}`);
            console.log(`Trigger Name: ${chalk_1.default.cyan(trigger.triggerName)}`);
            console.log(`TriggerConfig: ${chalk_1.default.magenta(JSON.stringify(trigger.triggerConfig, null, 2))}`);
            console.log(`Connection ID: ${chalk_1.default.yellow(trigger.connectionId)}`);
            console.log(""); // Add an empty line for better readability between triggers
        }
    }
}
exports.ActiveTriggers = ActiveTriggers;
