// composio-core/lib/src/env/base.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.RemoteWorkspace = exports.Workspace = exports.ShellFactory = void 0;
const uuid_1 = require("uuid");
const axios_1 = __importDefault(require("axios"));
const shared_1 = require("../utils/shared");
const ENV_GITHUB_ACCESS_TOKEN = "GITHUB_ACCESS_TOKEN";
const ENV_ACCESS_TOKEN = "ACCESS_TOKEN";
const ENV_COMPOSIO_API_KEY = "COMPOSIO_API_KEY";
const ENV_COMPOSIO_BASE_URL = "COMPOSIO_BASE_URL";
function _readEnvVar(name, defaultValue) {
    const value = (0, shared_1.getEnvVariable)(name, defaultValue) || defaultValue;
    if (value === undefined) {
        throw new Error(`Please provide value for \`${name}\``);
    }
    return value;
}
class Shell {
    constructor() {
        this._id = (0, uuid_1.v4)();
    }
    sanitizeCommand(cmd) {
        return `${cmd.trim()}\n`;
    }
    toString() {
        return `Shell(type=${this.constructor.name}, id=${this.id})`;
    }
    get id() {
        return this._id;
    }
    setup() {
        throw new Error("Method 'setup()' must be implemented.");
    }
    exec(cmd) {
        throw new Error("Method 'exec()' must be implemented.");
    }
    stop() {
        throw new Error("Method 'stop()' must be implemented.");
    }
}
class ShellFactory {
    constructor(factory) {
        this._factory = factory;
        this._recent = null;
        this._shells = {};
    }
    get recent() {
        return this._recent || this.new();
    }
    set recent(shell) {
        this._recent = shell;
    }
    new() {
        const shell = this._factory();
        shell.setup();
        this._shells[shell.id] = shell;
        this.recent = shell;
        return shell;
    }
    get(id = null) {
        if (!id) {
            return this.recent;
        }
        if (!this._shells[id]) {
            throw new Error(`No shell found with ID: ${id}`);
        }
        const shell = this._shells[id];
        this.recent = shell;
        return shell;
    }
    exec(cmd, id = null) {
        return this.get(id).exec(cmd);
    }
    stop(id) {
        if (!this._shells[id]) {
            return;
        }
        const shell = this._shells[id];
        shell.stop();
        delete this._shells[id];
    }
    teardown() {
        Object.keys(this._shells).forEach(id => {
            this._shells[id].stop();
            delete this._shells[id];
        });
        this._recent = null;
    }
}
exports.ShellFactory = ShellFactory;
class Workspace {
    constructor(configRepo) {
        this.id = (0, uuid_1.v4)();
        this.accessToken = (0, uuid_1.v4)().replace(/-/g, "");
        this.composioAPIKey = _readEnvVar(ENV_COMPOSIO_API_KEY, configRepo.config.composioAPIKey);
        this.composioBaseURL = _readEnvVar(ENV_COMPOSIO_BASE_URL, configRepo.config.composioBaseURL) + (configRepo.config.composioBaseURL.endsWith('/api') ? '' : '/api');
        this.githubAccessToken = configRepo.config.githubAccessToken || (0, shared_1.getEnvVariable)(ENV_GITHUB_ACCESS_TOKEN, "NO_VALUE");
        this.environment = {
            ...(configRepo.config.environment || {}),
            [ENV_COMPOSIO_API_KEY]: this.composioAPIKey,
            [ENV_COMPOSIO_BASE_URL]: this.composioBaseURL,
            [ENV_GITHUB_ACCESS_TOKEN]: this.githubAccessToken,
            [`_COMPOSIO_${ENV_GITHUB_ACCESS_TOKEN}`]: this.githubAccessToken,
            [ENV_ACCESS_TOKEN]: this.accessToken
        };
    }
    toString() {
        return `Workspace(type=${this.constructor.name}, id=${this.id})`;
    }
    setup() {
        throw new Error("Method 'setup()' must be implemented.");
    }
    get shells() {
        if (!this._shell_factory) {
            this._shell_factory = new ShellFactory(() => this._createShell());
        }
        return this._shell_factory;
    }
    _createShell() {
        throw new Error("Method '_create_shell()' must be implemented.");
    }
    executeAction(action, request_data, metadata = {}) {
        throw new Error("Method 'execute_action()' must be implemented.");
    }
    teardown() {
        this.shells.teardown();
    }
}
exports.Workspace = Workspace;
class RemoteWorkspace extends Workspace {
    constructor(configRepo) {
        super(configRepo);
        this.url = configRepo.config.composioBaseURL + (configRepo.config.composioBaseURL.endsWith('/api') ? '' : '/api');
    }
    async _request(endpoint, method, json = null, timeout = 60000.0) {
        return (0, axios_1.default)({
            url: `${this.url}${endpoint}`,
            method: method,
            data: json,
            headers: {
                "x-api-key": this.accessToken
            },
            timeout: timeout
        });
    }
    _createShell() {
        throw new Error("Creating shells for remote workspaces is not allowed.");
    }
    _upload(action) {
        throw new Error("Method '_upload()' must be implemented.");
    }
    async getLocalActionsSchema() {
        const request = await this._request("/local_actions", "get");
        return request.data.data;
    }
    async executeAction(action, request_data, metadata = {}) {
        var _a;
        if (!metadata.entityId) {
            metadata.entityId = "default";
        }
        const request = await this._request(`/actions/execute/${action}`, "post", {
            params: request_data,
            entity_id: metadata.entityId,
            metadata: metadata
        });
        const response = request.data;
        if (!response.error) {
            return response.data;
        }
        if ((_a = response.error) === null || _a === void 0 ? void 0 : _a.includes("Invalid value")) {
            return {
                status: "NOT_FOUND",
                error: `Action ${action} does not exist or supported.`,
                instructions: `Please only refer to the tools you have access to`
            };
        }
        throw new Error(`Error while executing ${action}: ${response.error}`);
    }
}
exports.RemoteWorkspace = RemoteWorkspace;
