// composio-core/lib/src/env/factory.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceFactory = exports.ExecEnv = void 0;
const workspace_1 = require("./e2b/workspace");
const workspace_2 = require("./docker/workspace");
const logger_1 = __importDefault(require("../utils/logger"));
var ExecEnv;
(function (ExecEnv) {
    ExecEnv["HOST"] = "HOST";
    ExecEnv["DOCKER"] = "DOCKER";
    ExecEnv["E2B"] = "E2B";
})(ExecEnv || (exports.ExecEnv = ExecEnv = {}));
class WorkspaceFactory {
    constructor(env, kwargs) {
        this.workspace = null;
        this.id = null;
        this.env = env;
        this.workspaceConfig = kwargs;
    }
    async new() {
        if (this.workspace) {
            return;
        }
        logger_1.default.debug(`Creating workspace with env=${this.workspaceConfig.env} and kwargs=${JSON.stringify(this.workspaceConfig.config)}`);
        let workspace = null;
        switch (this.workspaceConfig.env) {
            case ExecEnv.DOCKER:
                workspace = new workspace_2.DockerWorkspace(this.workspaceConfig);
                await workspace.setup();
                break;
            case ExecEnv.HOST:
                logger_1.default.warn("Local tools are not supported in host environment - see docs https://docs.composio.dev/swekit-js/workspace-env");
                break;
            case ExecEnv.E2B:
                workspace = new workspace_1.E2BWorkspace(this.workspaceConfig);
                await workspace.setup();
                break;
            default:
                throw new Error(`Unknown environment: ${this.workspaceConfig.env}`);
        }
        if (workspace) {
            this.workspace = workspace;
            this.id = workspace.id;
        }
    }
    async get(id = null) {
        return this.workspace;
    }
    async close() {
        var _a;
        await ((_a = this.workspace) === null || _a === void 0 ? void 0 : _a.teardown());
    }
}
exports.WorkspaceFactory = WorkspaceFactory;
