// composio-core/lib/src/utils/logger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const shared_1 = require("./shared");
const levels = {
    error: 'ERROR',
    warn: 'WARN',
    info: 'INFO',
    debug: 'DEBUG'
};
const colors = {
    red: (str) => `\x1b[31m${str}\x1b[0m`,
    yellow: (str) => `\x1b[33m${str}\x1b[0m`,
    blue: (str) => `\x1b[34m${str}\x1b[0m`,
    green: (str) => `\x1b[32m${str}\x1b[0m`,
    gray: (str) => `\x1b[90m${str}\x1b[0m`
};
const colorize = (level, timestamp) => {
    switch (level) {
        case 'error':
            return { level: colors.red(levels[level]), timestamp: colors.gray(timestamp) };
        case 'warn':
            return { level: colors.yellow(levels[level]), timestamp: colors.gray(timestamp) };
        case 'info':
            return { level: colors.blue(levels[level]), timestamp: colors.gray(timestamp) };
        case 'debug':
            return { level: colors.green(levels[level]), timestamp: colors.gray(timestamp) };
        default:
            return { level, timestamp };
    }
};
const logger = {
    level: (0, shared_1.getEnvVariable)("COMPOSIO_DEBUG", "0") === "1" ? 'debug' : 'info',
    log: (level, message, meta) => {
        const timestamp = new Date().toLocaleTimeString();
        const { level: coloredLevel, timestamp: coloredTimestamp } = colorize(level, timestamp);
        const metaInfo = meta ? ` - ${JSON.stringify(meta)}` : '';
        console.log(`[${coloredLevel}] ${coloredTimestamp} ${message}${metaInfo}`);
    },
    error: (message, meta) => logger.log('error', message, meta),
    warn: (message, meta) => logger.log('warn', message, meta),
    info: (message, meta) => logger.log('info', message, meta),
    debug: (message, meta) => logger.log('debug', message, meta)
};
exports.default = logger;
