// composio-core/lib/src/utils/shared.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.nodeExternalRequire = exports.getEnvVariable = exports.jsonSchemaToModel = exports.jsonSchemaToTsField = exports.jsonSchemaToTsType = void 0;
const zod_1 = __importDefault(require("zod"));
const PYDANTIC_TYPE_TO_TS_TYPE = {
    "string": String,
    "integer": Number,
    "number": Number,
    "boolean": Boolean,
    "null": null
};
function jsonSchemaToTsType(jsonSchema) {
    if (!jsonSchema.type) {
        jsonSchema.type = "string";
    }
    const type = jsonSchema.type;
    if (type === "array") {
        const itemsSchema = jsonSchema.items;
        if (itemsSchema) {
            const ItemType = jsonSchemaToTsType(itemsSchema);
            return ItemType;
        }
        return Array;
    }
    if (type === "object") {
        const properties = jsonSchema.properties;
        if (properties) {
            const nestedModel = jsonSchemaToModel(jsonSchema);
            return nestedModel;
        }
        return Object;
    }
    const tsType = PYDANTIC_TYPE_TO_TS_TYPE[type];
    if (tsType !== undefined) {
        return tsType;
    }
    throw new Error(`Unsupported JSON schema type: ${type}`);
}
exports.jsonSchemaToTsType = jsonSchemaToTsType;
function jsonSchemaToTsField(name, jsonSchema, required) {
    const description = jsonSchema.description;
    const examples = jsonSchema.examples || [];
    return [
        jsonSchemaToTsType(jsonSchema),
        {
            description: description,
            examples: examples,
            required: required.includes(name),
            default: required.includes(name) ? undefined : null,
        },
    ];
}
exports.jsonSchemaToTsField = jsonSchemaToTsField;
function jsonSchemaPropertiesToTSTypes(value) {
    if (!value.type) {
        return zod_1.default.object({});
    }
    let zodType;
    switch (value.type) {
        case "string":
            zodType = zod_1.default.string().describe((value.description || "") + (value.examples ? `\nExamples: ${value.examples.join(", ")}` : ""));
            break;
        case "number":
            zodType = zod_1.default.number().describe((value.description || "") + (value.examples ? `\nExamples: ${value.examples.join(", ")}` : ""));
            break;
        case "integer":
            zodType = zod_1.default.number().int().describe((value.description || "") + (value.examples ? `\nExamples: ${value.examples.join(", ")}` : ""));
            break;
        case "boolean":
            zodType = zod_1.default.boolean().describe((value.description || "") + (value.examples ? `\nExamples: ${value.examples.join(", ")}` : ""));
            break;
        case "array":
            zodType = zod_1.default.array(jsonSchemaPropertiesToTSTypes(value.items)).describe((value.description || "") + (value.examples ? `\nExamples: ${value.examples.join(", ")}` : ""));
            break;
        case "object":
            zodType = jsonSchemaToModel(value).describe((value.description || "") + (value.examples ? `\nExamples: ${value.examples.join(", ")}` : ""));
            break;
        case "null":
            zodType = zod_1.default.null().describe(value.description || "");
            break;
        default:
            throw new Error(`Unsupported JSON schema type: ${value.type}`);
    }
    return zodType;
}
function jsonSchemaToModel(jsonSchema) {
    const properties = jsonSchema.properties;
    const requiredFields = jsonSchema.required || [];
    if (!properties) {
        return zod_1.default.object({});
    }
    const zodSchema = {};
    for (const [key, _] of Object.entries(properties)) {
        const value = _;
        let zodType;
        if (value.anyOf) {
            const anyOfTypes = value.anyOf.map((schema) => jsonSchemaPropertiesToTSTypes(schema));
            zodType = zod_1.default.union(anyOfTypes).describe((value.description || "") + (value.examples ? `\nExamples: ${value.examples.join(", ")}` : ""));
        }
        else if (value.allOf) {
            const allOfTypes = value.allOf.map((schema) => jsonSchemaPropertiesToTSTypes(schema));
            zodType = zod_1.default.intersection(allOfTypes[0], allOfTypes.slice(1).reduce((acc, schema) => acc.and(schema), allOfTypes[0])).describe((value.description || "") + (value.examples ? `\nExamples: ${value.examples.join(", ")}` : ""));
        }
        else if (value.type) {
            zodType = jsonSchemaPropertiesToTSTypes(value);
        }
        else {
            throw new Error(`Missing 'type' property in JSON schema for key: ${key}`);
        }
        if (value.description) {
            zodType = zodType.describe(value.description);
        }
        if (requiredFields.includes(key)) {
            zodSchema[key] = zodType;
        }
        else {
            zodSchema[key] = zodType.optional();
        }
    }
    return zod_1.default.object(zodSchema);
}
exports.jsonSchemaToModel = jsonSchemaToModel;
const getEnvVariable = (name, defaultValue = undefined) => {
    try {
        return process.env[name] || defaultValue;
    }
    catch (e) {
        return defaultValue;
    }
};
exports.getEnvVariable = getEnvVariable;
const nodeExternalRequire = (name) => {
    try {
        if (typeof process !== 'undefined') {
            return require(name);
        }
        else {
            return require(`external:${name}`);
        }
    }
    catch (err) {
        return null;
    }
};
exports.nodeExternalRequire = nodeExternalRequire;
