// composio-core/lib/src/sdk/base.toolset.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ComposioToolSet = void 0;
const sdk_1 = require("../sdk");
const factory_1 = require("../env/factory");
const OpenAPI_1 = require("./client/core/OpenAPI");
const shared_1 = require("../utils/shared");
const env_1 = require("../env");
const logger_1 = __importDefault(require("../utils/logger"));
const fileUtils_1 = require("./utils/fileUtils");
const utils_1 = require("./utils");
class UserData {
    constructor(_path) {
        this._path = _path;
    }
    init() {
        try {
            const module = require(this._path);
            this.apiKey = module.apiKey;
        }
        catch (_a) {
            return false;
        }
    }
    static load(_path) {
        return new UserData(_path);
    }
}
const getUserPath = () => {
    try {
        const path = require("path");
        return path.join((0, shared_1.getEnvVariable)("HOME", ""), ".composio", "userData.json");
    }
    catch (_a) {
        return null;
    }
};
class ComposioToolSet {
    constructor(apiKey, baseUrl = OpenAPI_1.COMPOSIO_BASE_URL, runtime = null, entityId = "default", workspaceConfig = env_1.Workspace.Host()) {
        const clientApiKey = apiKey || (0, shared_1.getEnvVariable)("COMPOSIO_API_KEY") || UserData.load(getUserPath()).apiKey;
        if (!clientApiKey) {
            throw new Error("API key is required, please pass it either by using `COMPOSIO_API_KEY` environment variable or during initialization");
        }
        this.apiKey = clientApiKey;
        this.client = new sdk_1.Composio(this.apiKey, baseUrl || undefined, runtime);
        this.runtime = runtime;
        this.entityId = entityId;
        if (!workspaceConfig.config.composioBaseURL) {
            workspaceConfig.config.composioBaseURL = baseUrl;
        }
        if (!workspaceConfig.config.composioAPIKey) {
            workspaceConfig.config.composioAPIKey = apiKey;
        }
        this.workspace = new factory_1.WorkspaceFactory(workspaceConfig.env, workspaceConfig);
        this.workspaceEnv = workspaceConfig.env;
        if (typeof process !== 'undefined') {
            process.on("exit", async () => {
                var _a;
                await ((_a = this.workspace.workspace) === null || _a === void 0 ? void 0 : _a.teardown());
            });
        }
    }
    async setup() {
        await this.workspace.new();
        if (!this.localActions && this.workspaceEnv !== factory_1.ExecEnv.HOST) {
            this.localActions = await this.workspace.workspace.getLocalActionsSchema();
        }
    }
    async getActionsSchema(filters = {}, entityId) {
        var _a, _b;
        await this.setup();
        let actions = (await this.client.actions.list({
            actions: (_a = filters.actions) === null || _a === void 0 ? void 0 : _a.join(","),
            showAll: true
        })).items;
        const localActionsMap = new Map();
        (_b = filters.actions) === null || _b === void 0 ? void 0 : _b.forEach((action) => {
            var _a;
            const actionData = (_a = this.localActions) === null || _a === void 0 ? void 0 : _a.find((a) => a.name === action);
            if (actionData) {
                localActionsMap.set(actionData.name, actionData);
            }
        });
        const uniqueLocalActions = Array.from(localActionsMap.values());
        const toolsActions = [...actions, ...uniqueLocalActions];
        return toolsActions.map((action) => {
            return this.modifyActionForLocalExecution(action);
        });
    }
    async getToolsSchema(filters, entityId) {
        var _a, _b, _c, _d;
        await this.setup();
        const apps = await this.client.actions.list({
            ...((filters === null || filters === void 0 ? void 0 : filters.apps) && { apps: (_a = filters === null || filters === void 0 ? void 0 : filters.apps) === null || _a === void 0 ? void 0 : _a.join(",") }),
            ...((filters === null || filters === void 0 ? void 0 : filters.tags) && { tags: (_b = filters === null || filters === void 0 ? void 0 : filters.tags) === null || _b === void 0 ? void 0 : _b.join(",") }),
            ...((filters === null || filters === void 0 ? void 0 : filters.useCase) && { useCase: filters === null || filters === void 0 ? void 0 : filters.useCase }),
            ...((filters === null || filters === void 0 ? void 0 : filters.actions) && { actions: (_c = filters === null || filters === void 0 ? void 0 : filters.actions) === null || _c === void 0 ? void 0 : _c.join(",") }),
        });
        const localActions = new Map();
        if (filters.apps && Array.isArray(filters.apps)) {
            for (const appName of filters.apps) {
                const actionData = (_d = this.localActions) === null || _d === void 0 ? void 0 : _d.filter((a) => a.appName === appName);
                if (actionData) {
                    for (const action of actionData) {
                        localActions.set(action.name, action);
                    }
                }
            }
        }
        const uniqueLocalActions = Array.from(localActions.values());
        const toolsActions = [...apps.items, ...uniqueLocalActions];
        return toolsActions.map((action) => {
            return this.modifyActionForLocalExecution(action);
        });
    }
    modifyActionForLocalExecution(toolSchema) {
        const properties = (0, utils_1.convertReqParams)(toolSchema.parameters.properties);
        toolSchema.parameters.properties = properties;
        const response = toolSchema.response.properties;
        for (const responseKey of Object.keys(response)) {
            if (responseKey === "file") {
                response["file_uri_path"] = {
                    type: "string",
                    title: "Name",
                    description: "Local absolute path to the file or http url to the file"
                };
                delete response[responseKey];
            }
        }
        return toolSchema;
    }
    async getActions(filters = {}, entityId) {
        throw new Error("Not implemented");
    }
    async getTools(filters, entityId) {
        throw new Error("Not implemented");
    }
    async executeAction(action, params, entityId = "default") {
        if (this.workspaceEnv && this.workspaceEnv !== factory_1.ExecEnv.HOST) {
            const workspace = await this.workspace.get();
            return workspace.executeAction(action, params, {
                entityId: this.entityId
            });
        }
        params = await (0, utils_1.converReqParamForActionExecution)(params);
        const data = await this.client.getEntity(entityId).execute(action, params);
        return this.processResponse(data, {
            action: action,
            entityId: entityId
        });
    }
    async processResponse(data, meta) {
        var _a;
        const isFile = !!((_a = data === null || data === void 0 ? void 0 : data.response_data) === null || _a === void 0 ? void 0 : _a.file);
        if (isFile) {
            const fileData = data.response_data.file;
            const { name, content } = fileData;
            const file_name_prefix = `${meta.action}_${meta.entityId}_${Date.now()}`;
            const filePath = (0, fileUtils_1.saveFile)(file_name_prefix, content);
            delete data.response_data.file;
            return {
                ...data,
                response_data: {
                    ...data.response_data,
                    file_uri_path: filePath
                }
            };
        }
        return data;
    }
    async execute_action(action, 
    // this need to improve
    params, entityId = "default") {
        logger_1.default.warn("execute_action is deprecated, use executeAction instead");
        return this.executeAction(action, params, entityId);
    }
}
exports.ComposioToolSet = ComposioToolSet;
