// composio-core/lib/src/sdk/base.toolset.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const globals_1 = require("@jest/globals");
const base_toolset_1 = require("./base.toolset");
const getTestConfig_1 = require("../../config/getTestConfig");
(0, globals_1.describe)("ComposioToolSet class tests", () => {
    let toolset;
    const testConfig = (0, getTestConfig_1.getTestConfig)();
    (0, globals_1.beforeAll)(() => {
        toolset = new base_toolset_1.ComposioToolSet(testConfig.COMPOSIO_API_KEY, testConfig.BACKEND_HERMES_URL);
    });
    (0, globals_1.it)("should create a ComposioToolSet instance", async () => {
        const tools = await toolset.getToolsSchema({ apps: ["github"] });
        (0, globals_1.expect)(tools).toBeInstanceOf(Array);
        (0, globals_1.expect)(tools).not.toHaveLength(0);
    });
    (0, globals_1.it)("should create a ComposioToolSet instance with apps and tags", async () => {
        const tools = await toolset.getToolsSchema({ apps: ["github"], tags: ["important"] });
        (0, globals_1.expect)(tools).toBeInstanceOf(Array);
        (0, globals_1.expect)(tools).not.toHaveLength(0);
    });
    (0, globals_1.it)("should create a ComposioToolSet instance with actions", async () => {
        const tools = await toolset.getActionsSchema({ actions: ["github_issues_create"] });
        (0, globals_1.expect)(tools).toBeInstanceOf(Array);
    });
    (0, globals_1.it)("should execute an action", async () => {
        const actionName = "github_issues_create";
        const requestBody = {
            owner: "utkarsh-dixit",
            repo: "speedy",
            title: "Test issue",
            body: "This is a test issue",
            appNames: "github"
        };
        const executionResult = await toolset.executeAction(actionName, requestBody, "default");
        (0, globals_1.expect)(executionResult).toBeDefined();
        // @ts-ignore
        (0, globals_1.expect)(executionResult).toHaveProperty('successfull', true);
        (0, globals_1.expect)(executionResult.data).toBeDefined();
    });
    (0, globals_1.it)("should execute an file upload", async () => {
        const ACTION_NAME = "GMAIL_SEND_EMAIL";
        const actions = await toolset.getToolsSchema({ actions: [ACTION_NAME] });
        // Check if exist
        (0, globals_1.expect)(actions[0].parameters.properties["attachment_file_uri_path"]).toBeDefined();
        const requestBody = {
            recipient_email: "himanshu@composio.dev",
            subject: "Test email from himanshu",
            body: "This is a test email",
            attachment_file_uri_path: "https://composio.dev/wp-content/uploads/2024/07/Composio-Logo.webp"
        };
        const executionResult = await toolset.executeAction(ACTION_NAME, requestBody, "default");
        (0, globals_1.expect)(executionResult).toBeDefined();
        // @ts-ignore
        (0, globals_1.expect)(executionResult).toHaveProperty('successfull', true);
        (0, globals_1.expect)(executionResult.data).toBeDefined();
    });
});
