// composio-core/lib/src/sdk/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Composio = void 0;
const connectedAccounts_1 = require("./models/connectedAccounts");
const apps_1 = require("./models/apps");
const actions_1 = require("./models/actions");
const triggers_1 = require("./models/triggers");
const integrations_1 = require("./models/integrations");
const activeTriggers_1 = require("./models/activeTriggers");
const backendClient_1 = require("./models/backendClient");
const Entity_1 = require("./models/Entity");
const axios_1 = __importDefault(require("axios"));
const projectUtils_1 = require("./utils/projectUtils");
const other_1 = require("./utils/other");
const config_1 = require("./utils/config");
class Composio {
    /**
     * Initializes a new instance of the Composio class.
     *
     * @param {string} [apiKey] - The API key for authenticating with the Composio backend. Can also be set locally in an environment variable.
     * @param {string} [baseUrl] - The base URL for the Composio backend. By default, it is set to the production URL.
     * @param {string} [runtime] - The runtime environment for the SDK.
     */
    constructor(apiKey, baseUrl, runtime) {
        // // Parse the base URL and API key, falling back to environment variables or defaults if not provided.
        const { baseURL: baseURLParsed, apiKey: apiKeyParsed } = (0, config_1.getClientBaseConfig)(baseUrl, apiKey);
        // Initialize the BackendClient with the parsed API key and base URL.
        this.backendClient = new backendClient_1.BackendClient(apiKeyParsed, baseURLParsed, runtime);
        // Instantiate models with dependencies as needed.
        this.connectedAccounts = new connectedAccounts_1.ConnectedAccounts(this.backendClient);
        this.triggers = new triggers_1.Triggers(this.backendClient);
        this.apps = new apps_1.Apps(this.backendClient);
        this.actions = new actions_1.Actions(this.backendClient);
        this.integrations = new integrations_1.Integrations(this.backendClient);
        this.activeTriggers = new activeTriggers_1.ActiveTriggers(this.backendClient);
        this.checkForLatestVersionFromNPM();
    }
    /**
     * Checks for the latest version of the Composio SDK from NPM.
     * If a newer version is available, it logs a warning to the console.
     */
    async checkForLatestVersionFromNPM() {
        try {
            const packageName = "composio-core";
            const packageJsonDir = (0, projectUtils_1.getPackageJsonDir)();
            const currentVersionFromPackageJson = require(packageJsonDir + '/package.json').version;
            const response = await axios_1.default.get(`https://registry.npmjs.org/${packageName}/latest`);
            const latestVersion = response.data.version;
            if ((0, other_1.isNewerVersion)(latestVersion, currentVersionFromPackageJson)) {
                console.warn(`🚀 Upgrade available! Your composio-core version (${currentVersionFromPackageJson}) is behind. Latest version: ${latestVersion}.`);
            }
        }
        catch (error) {
            // Ignore and do nothing
        }
    }
    /**
     * Retrieves an Entity instance associated with a given ID.
     *
     * @param {string} [id='default'] - The ID of the entity to retrieve.
     * @returns {Entity} An instance of the Entity class.
     */
    getEntity(id = 'default') {
        return new Entity_1.Entity(this.backendClient, id);
    }
}
exports.Composio = Composio;
