// composio-core/lib/src/sdk/utils.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.converReqParamForActionExecution = exports.convertReqParams = void 0;
const axios_1 = __importDefault(require("axios"));
const fs_1 = __importDefault(require("fs"));
const convertReqParams = (properties) => {
    var _a, _b;
    const clonedProperties = JSON.parse(JSON.stringify(properties));
    for (const propertyKey of Object.keys(clonedProperties)) {
        const object = clonedProperties[propertyKey];
        const isObject = typeof object === "object";
        const isFile = isObject && (((_a = object === null || object === void 0 ? void 0 : object.required) === null || _a === void 0 ? void 0 : _a.includes("name")) && ((_b = object === null || object === void 0 ? void 0 : object.required) === null || _b === void 0 ? void 0 : _b.includes("content")));
        if (isFile) {
            const newKey = `${propertyKey}_file_uri_path`;
            clonedProperties[newKey] = {
                type: "string",
                title: "Name",
                description: "Local absolute path to the file or http url to the file"
            };
            delete clonedProperties[propertyKey];
        }
    }
    return clonedProperties;
};
exports.convertReqParams = convertReqParams;
const converReqParamForActionExecution = async (params) => {
    const clonedParams = JSON.parse(JSON.stringify(params));
    for (const key of Object.keys(clonedParams)) {
        if (key.includes("file_uri_path")) {
            const initKey = key.replace("_file_uri_path", "");
            clonedParams[initKey] = {};
            const newKey = clonedParams[initKey];
            const valuePassedByClient = clonedParams[key];
            // if not http url
            if (valuePassedByClient.startsWith("http") || valuePassedByClient.startsWith("https")) {
                newKey.name = valuePassedByClient.split("/").pop();
                const response = await axios_1.default.get(valuePassedByClient, { responseType: 'arraybuffer' });
                newKey.content = Buffer.from(response.data, 'binary').toString('base64');
            }
            else {
                const file = await fs_1.default.promises.readFile(valuePassedByClient, {
                    encoding: null
                });
                newKey.content = file.toString('base64');
                newKey.name = valuePassedByClient.split("/").pop();
            }
            delete clonedParams[key];
        }
    }
    return clonedParams;
};
exports.converReqParamForActionExecution = converReqParamForActionExecution;
