// composio-core/lib/src/frameworks/cloudflare.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudflareToolSet = void 0;
const base_toolset_1 = require("../sdk/base.toolset");
const OpenAPI_1 = require("../sdk/client/core/OpenAPI");
const env_1 = require("../env");
const logger_1 = __importDefault(require("../utils/logger"));
class CloudflareToolSet extends base_toolset_1.ComposioToolSet {
    /**
     * Composio toolset for Cloudflare framework.
     *
     * Example:
     * ```typescript
     *
     * ```
     */
    constructor(config = {}) {
        super(config.apiKey || null, config.baseUrl || OpenAPI_1.COMPOSIO_BASE_URL, "cloudflare", config.entityId || "default", config.workspaceConfig || env_1.Workspace.Host());
    }
    /**
     * @deprecated Use getTools instead.
     */
    async getActions(filters) {
        const actions = await this.getActionsSchema(filters);
        return actions.map((action) => {
            const formattedSchema = {
                name: action.name,
                description: action.description,
                parameters: action.parameters,
            };
            const tool = {
                type: "function",
                function: formattedSchema,
            };
            return tool;
        }) || [];
    }
    /**
     * @deprecated Use getActions instead.
     */
    async get_actions(filters) {
        logger_1.default.warn("get_actions is deprecated, use getActions instead");
        return this.getActions(filters);
    }
    async getTools(filters) {
        const actions = await this.getToolsSchema(filters);
        return actions.map((action) => {
            const formattedSchema = {
                name: action.name,
                description: action.description,
                parameters: action.parameters,
            };
            const tool = {
                type: "function",
                function: formattedSchema,
            };
            return tool;
        }) || [];
    }
    /**
     * @deprecated Use getTools instead.
     */
    async get_tools(filters) {
        logger_1.default.warn("get_tools is deprecated, use getTools instead");
        return this.getTools(filters);
    }
    async executeToolCall(tool, entityId = null) {
        return JSON.stringify(await this.execute_action(tool.name, typeof tool.arguments === "string" ? JSON.parse(tool.arguments) : tool.arguments, entityId || this.entityId));
    }
    /**
     * @deprecated Use executeToolCall instead.
     */
    async execute_tool_call(tool, entityId = null) {
        logger_1.default.warn("execute_tool_call is deprecated, use executeToolCall instead");
        return this.executeToolCall(tool, entityId);
    }
    async handleToolCall(result, entityId = null) {
        const outputs = [];
        if ("tool_calls" in result && Array.isArray(result.tool_calls)) {
            for (const tool_call of result.tool_calls) {
                if (tool_call.name) {
                    outputs.push(await this.executeToolCall(tool_call, entityId));
                }
            }
        }
        return outputs;
    }
    /**
     * @deprecated Use handleToolCall instead.
     */
    async handle_tool_call(result, entityId = null) {
        logger_1.default.warn("handle_tool_call is deprecated, use handleToolCall instead");
        return this.handleToolCall(result, entityId);
    }
}
exports.CloudflareToolSet = CloudflareToolSet;
