// composio-core/lib/src/frameworks/langchain.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.LangchainToolSet = void 0;
const base_toolset_1 = require("../sdk/base.toolset");
const shared_1 = require("../utils/shared");
const tools_1 = require("@langchain/core/tools");
const OpenAPI_1 = require("../sdk/client/core/OpenAPI");
const env_1 = require("../env");
const logger_1 = __importDefault(require("../utils/logger"));
class LangchainToolSet extends base_toolset_1.ComposioToolSet {
    /**
     * Composio toolset for Langchain framework.
     *
     * Example:
     * ```typescript
     * import * as dotenv from "dotenv";
     * import { App, ComposioToolSet } from "composio_langchain";
     * import { AgentExecutor, create_openai_functions_agent } from "langchain/agents";
     * import { ChatOpenAI } from "langchain_openai";
     * import { hub } from "langchain";
     *
     * // Load environment variables from .env
     * dotenv.config();
     *
     * // Pull relevant agent model.
     * const prompt = hub.pull("hwchase17/openai-functions-agent");
     *
     * // Initialize tools.
     * const openai_client = new ChatOpenAI({ apiKey: process.env.OPENAI_API_KEY });
     * const composio_toolset = new ComposioToolSet();
     *
     * // Get All the tools
     * const tools = composio_toolset.get_tools({ apps: [App.GITHUB] });
     *
     * // Define task
     * const task = "Star a repo composiohq/composio on GitHub";
     *
     * // Define agent
     * const agent = create_openai_functions_agent(openai_client, tools, prompt);
     * const agent_executor = new AgentExecutor({ agent, tools, verbose: true });
     *
     * // Execute using agent_executor
     * agent_executor.invoke({ input: task });
     * ```
     */
    constructor(config = {}) {
        super(config.apiKey || null, config.baseUrl || OpenAPI_1.COMPOSIO_BASE_URL, "langchain", config.entityId || "default", config.workspaceConfig || env_1.Workspace.Host());
    }
    _wrapTool(schema, entityId = null) {
        const app = schema["appName"];
        const action = schema["name"];
        const description = schema["description"];
        const func = async (...kwargs) => {
            return JSON.stringify(await this.executeAction(action, kwargs[0], entityId || this.entityId));
        };
        const parameters = (0, shared_1.jsonSchemaToModel)(schema["parameters"]);
        // @TODO: Add escriiption an othjer stuff here
        return new tools_1.DynamicStructuredTool({
            name: action,
            description,
            schema: parameters,
            func: func
        });
    }
    /**
        * @deprecated Use getTools instead.
    */
    async getActions(filters = {}, entityId) {
        const actions = await this.getActionsSchema(filters, entityId);
        return actions.map((tool) => this._wrapTool(tool, entityId || this.entityId));
    }
    /**
     * @deprecated Use getTools instead.
     */
    async get_actions(filters = {}, entityId) {
        logger_1.default.warn("get_actions is deprecated, use getActions instead");
        return this.getActions(filters, entityId);
    }
    async getTools(filters, entityId = null) {
        const tools = await this.getToolsSchema(filters, entityId);
        return tools.map((tool) => this._wrapTool(tool, entityId || this.entityId));
    }
    /**
     * @deprecated Use getTools instead.
     */
    async get_tools(filters, entityId) {
        logger_1.default.warn("get_tools is deprecated, use getTools instead");
        return this.getTools(filters, entityId);
    }
}
exports.LangchainToolSet = LangchainToolSet;
