// composio-core/lib/src/frameworks/openai.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.OpenAIToolSet = void 0;
const base_toolset_1 = require("../sdk/base.toolset");
const OpenAPI_1 = require("../sdk/client/core/OpenAPI");
const env_1 = require("../env");
const logger_1 = __importDefault(require("../utils/logger"));
class OpenAIToolSet extends base_toolset_1.ComposioToolSet {
    /**
     * Composio toolset for OpenAI framework.
     *
     * Example:
     * ```typescript
     *
     * ```
     */
    constructor(config = {}) {
        super(config.apiKey || null, config.baseUrl || OpenAPI_1.COMPOSIO_BASE_URL, "openai", config.entityId || "default", config.workspaceConfig || env_1.Workspace.Host());
    }
    /**
     * @deprecated Use getTools instead.
     */
    async getActions(filters = {}, entityId) {
        const mainActions = await this.getActionsSchema(filters, entityId);
        return mainActions.map((action) => {
            const formattedSchema = {
                name: action.name,
                description: action.description,
                parameters: action.parameters
            };
            const tool = {
                type: "function",
                function: formattedSchema
            };
            return tool;
        }) || [];
    }
    /**
     * @deprecated Use getTools instead.
     */
    async get_actions(filters = {}, entityId) {
        logger_1.default.warn("get_actions is deprecated, use getActions instead");
        return this.getActions(filters, entityId);
    }
    async getTools(filters, entityId) {
        const mainActions = await this.getToolsSchema(filters, entityId);
        return mainActions.map((action) => {
            const formattedSchema = {
                name: action.name,
                description: action.description,
                parameters: action.parameters,
            };
            const tool = {
                type: "function",
                function: formattedSchema
            };
            return tool;
        }) || [];
    }
    /**
     * @deprecated Use getTools instead.
     */
    async get_tools(filters, entityId) {
        logger_1.default.warn("get_tools is deprecated, use getTools instead");
        return this.getTools(filters, entityId);
    }
    async executeToolCall(tool, entityId = null) {
        return JSON.stringify(await this.executeAction(tool.function.name, JSON.parse(tool.function.arguments), entityId || this.entityId));
    }
    /**
     * @deprecated Use executeToolCall instead.
     */
    async execute_tool_call(tool, entityId = null) {
        logger_1.default.warn("execute_tool_call is deprecated, use executeToolCall instead");
        return this.executeToolCall(tool, entityId);
    }
    async handleToolCall(chatCompletion, entityId = null) {
        const outputs = [];
        for (const message of chatCompletion.choices) {
            if (message.message.tool_calls) {
                outputs.push(await this.executeToolCall(message.message.tool_calls[0], entityId));
            }
        }
        return outputs;
    }
    /**
     * @deprecated Use handleToolCall instead.
     */
    async handle_tool_call(chatCompletion, entityId = null) {
        logger_1.default.warn("handle_tool_call is deprecated, use handleToolCall instead");
        return this.handleToolCall(chatCompletion, entityId);
    }
    async handleAssistantMessage(run, entityId = null) {
        var _a, _b;
        const tool_calls = ((_b = (_a = run.required_action) === null || _a === void 0 ? void 0 : _a.submit_tool_outputs) === null || _b === void 0 ? void 0 : _b.tool_calls) || [];
        const tool_outputs = await Promise.all(tool_calls.map(async (tool_call) => {
            logger_1.default.debug(`Executing tool call with ID: ${tool_call.function.name} and parameters: ${JSON.stringify(tool_call.function.arguments)}`);
            const tool_response = await this.executeToolCall(tool_call, entityId || this.entityId);
            logger_1.default.debug(`Received tool response: ${JSON.stringify(tool_response)}`);
            return {
                tool_call_id: tool_call.id,
                output: JSON.stringify(tool_response),
            };
        }));
        return tool_outputs;
    }
    /**
     * @deprecated Use handleAssistantMessage instead.
     */
    async handle_assistant_message(run, entityId = null) {
        logger_1.default.warn("handle_assistant_message is deprecated, use handleAssistantMessage instead");
        return this.handleAssistantMessage(run, entityId);
    }
    async waitAndHandleAssistantToolCalls(client, run, thread, entityId = null) {
        while (["queued", "in_progress", "requires_action"].includes(run.status)) {
            logger_1.default.debug(`Current run status: ${run.status}`);
            const tool_outputs = await this.handleAssistantMessage(run, entityId || this.entityId);
            if (run.status === "requires_action") {
                logger_1.default.debug(`Submitting tool outputs for run ID: ${run.id} in thread ID: ${thread.id}`);
                run = await client.beta.threads.runs.submitToolOutputs(thread.id, run.id, {
                    tool_outputs: tool_outputs
                });
            }
            else {
                run = await client.beta.threads.runs.retrieve(thread.id, run.id);
                await new Promise(resolve => setTimeout(resolve, 500));
            }
        }
        return run;
    }
    /**
     * @deprecated Use waitAndHandleAssistantToolCalls instead.
     */
    async wait_and_handle_assistant_tool_calls(client, run, thread, entityId = null) {
        logger_1.default.warn("wait_and_handle_assistant_tool_calls is deprecated, use waitAndHandleAssistantToolCalls instead");
        return this.waitAndHandleAssistantToolCalls(client, run, thread, entityId);
    }
}
exports.OpenAIToolSet = OpenAIToolSet;
