// composio-core/lib/src/frameworks/vercel.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VercelAIToolSet = void 0;
const ai_1 = require("ai");
const base_toolset_1 = require("../sdk/base.toolset");
const shared_1 = require("../utils/shared");
class VercelAIToolSet extends base_toolset_1.ComposioToolSet {
    constructor(config = {}) {
        super(config.apiKey || null, config.baseUrl || null, "vercel-ai", config.entityId || "default");
    }
    generateVercelTool(schema) {
        const parameters = (0, shared_1.jsonSchemaToModel)(schema.parameters);
        return (0, ai_1.tool)({
            description: schema.description,
            parameters,
            execute: async (params) => {
                return await this.execute_tool_call({
                    name: schema.name,
                    arguments: JSON.stringify(params)
                }, this.entityId);
            },
        });
    }
    async get_actions(filters) {
        const actionsList = await this.getActionsSchema(filters);
        const tools = {};
        actionsList.forEach(actionSchema => {
            // @ts-ignore
            tools[actionSchema.name] = this.generateVercelTool(actionSchema);
        });
        return tools;
    }
    // change this implementation
    async get_tools(filters) {
        var _a, _b, _c, _d, _e;
        const actionsList = await this.client.actions.list({
            ...((filters === null || filters === void 0 ? void 0 : filters.apps) && { apps: (_a = filters === null || filters === void 0 ? void 0 : filters.apps) === null || _a === void 0 ? void 0 : _a.join(",") }),
            ...((filters === null || filters === void 0 ? void 0 : filters.tags) && { tags: (_b = filters === null || filters === void 0 ? void 0 : filters.tags) === null || _b === void 0 ? void 0 : _b.join(",") }),
            ...((filters === null || filters === void 0 ? void 0 : filters.useCase) && { useCase: filters === null || filters === void 0 ? void 0 : filters.useCase }),
            ...((filters === null || filters === void 0 ? void 0 : filters.actions) && { actions: (_c = filters === null || filters === void 0 ? void 0 : filters.actions) === null || _c === void 0 ? void 0 : _c.join(",") }),
            ...((filters === null || filters === void 0 ? void 0 : filters.actions) && { actions: (_d = filters === null || filters === void 0 ? void 0 : filters.actions) === null || _d === void 0 ? void 0 : _d.join(",") }),
        });
        const tools = {};
        (_e = actionsList.items) === null || _e === void 0 ? void 0 : _e.forEach(actionSchema => {
            // @ts-ignore
            tools[actionSchema.name] = this.generateVercelTool(actionSchema);
        });
        return tools;
    }
    async getTools(filters) {
        return await this.get_tools(filters);
    }
    async execute_tool_call(tool, entityId = null) {
        return JSON.stringify(await this.executeAction(tool.name, typeof tool.arguments === "string" ? JSON.parse(tool.arguments) : tool.arguments, entityId || this.entityId));
    }
}
exports.VercelAIToolSet = VercelAIToolSet;
