// composio-core/lib/src/sdk/client/services.gen.js
"use strict";
// This file is auto-generated by @hey-api/openapi-ts
Object.defineProperty(exports, "__esModule", { value: true });
exports.ActionsControllerV2Service = exports.LogsService = exports.MetadataService = exports.CliService = exports.AdminService = exports.TriggersService = exports.AppLoginService = exports.ConnectionsService = exports.ActionsControllerV1Service = exports.AppConnectorService = exports.AppService = exports.ApiKeysService = exports.TeamService = exports.ClientAuthService = exports.client = void 0;
const client_axios_1 = require("@hey-api/client-axios");
exports.client = (0, client_axios_1.createClient)((0, client_axios_1.createConfig)());
class ClientAuthService {
    /**
     * Identify client
     */
    static identifyClient(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/client/auth/identify'
        });
    }
    /**
     * Track client
     */
    static trackClient(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/client/auth/track'
        });
    }
    /**
     * Send magic link
     */
    static sendMagicLink(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/client/auth/send_magic_link'
        });
    }
    /**
     * Logout API
     * Logout the user and clear the server side session
     */
    static logout(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/client/auth/logout'
        });
    }
    /**
     * Verify magic link
     */
    static verifyMagicLink(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/client/auth/verify_magic_link'
        });
    }
    /**
     * Get user info
     */
    static getUserInfo(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/client/auth/client_info'
        });
    }
    /**
     * Pusher user auth
     */
    static pusherUserAuth(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/client/auth/pusher_auth'
        });
    }
    /**
     * Delete client
     */
    static deleteClient(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).delete({
            ...options,
            url: '/api/v1/client/auth/delete/{clientId}'
        });
    }
}
exports.ClientAuthService = ClientAuthService;
class TeamService {
    /**
     * Invite member
     */
    static inviteMember(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/team/invite'
        });
    }
    /**
     * List members
     */
    static listMembers(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/team/members'
        });
    }
    /**
     * Remove member
     */
    static removeMember(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).delete({
            ...options,
            url: '/api/v1/team/members/{id}'
        });
    }
}
exports.TeamService = TeamService;
class ApiKeysService {
    /**
     * Generate api key
     */
    static generateApiKey(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/api_keys'
        });
    }
    /**
     * List api keys
     */
    static listApiKeys(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/api_keys'
        });
    }
    /**
     * Delete api key
     */
    static deleteApiKey(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).delete({
            ...options,
            url: '/api/v1/api_keys/{id}'
        });
    }
}
exports.ApiKeysService = ApiKeysService;
class AppService {
    /**
     * Get apps
     * Retrieve a list of all applications based on query parameters.
     */
    static getApps(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/apps'
        });
    }
    /**
     * Add open api spec
     * Add openapi spec tool
     */
    static addOpenApiSpec(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/apps/openapi/spec/add'
        });
    }
    /**
     * Get opena api specs
     */
    static getOpenaApiSpecs(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/apps/openapi/spec/list'
        });
    }
    /**
     * Delete open api spec tool
     */
    static deleteOpenApiSpecTool(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).delete({
            ...options,
            url: '/api/v1/apps/openapi/spec/delete'
        });
    }
    /**
     * Get open api spec logs
     */
    static getOpenApiSpecLogs(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/apps/openapi/spec/logs/{name}'
        });
    }
    /**
     * Add custom repo
     */
    static addCustomRepo(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/apps/custom-repo/add-repo'
        });
    }
    /**
     * Get app
     * Get app details
     */
    static getApp(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/apps/{appName}'
        });
    }
    /**
     * Sync apps
     * Sync custom repo
     */
    static syncApps(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/apps/sync'
        });
    }
    /**
     * Get custom repo
     * Get custom repo
     */
    static getCustomRepo(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/apps/custom-repo/get-repo'
        });
    }
}
exports.AppService = AppService;
class AppConnectorService {
    /**
     * Create connector
     */
    static createConnector(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/integrations'
        });
    }
    /**
     * List global connectors
     */
    static listGlobalConnectors(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/integrations'
        });
    }
    /**
     * Get connector info
     */
    static getConnectorInfo(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/integrations/{integrationId}'
        });
    }
    /**
     * Patch connector
     */
    static patchConnector(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).patch({
            ...options,
            url: '/api/v1/integrations/{integrationId}'
        });
    }
    /**
     * Delete connector
     */
    static deleteConnector(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).delete({
            ...options,
            url: '/api/v1/integrations/{integrationId}'
        });
    }
    /**
     * Patch post connector
     */
    static patchPostConnector(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/integrations/{integrationId}/status'
        });
    }
}
exports.AppConnectorService = AppConnectorService;
class ActionsControllerV1Service {
    /**
     * List actions
     */
    static v1ListActions(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/actions'
        });
    }
    /**
     * Execute action
     */
    static v1ExecuteAction(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/actions/{actionId}/execute'
        });
    }
    /**
     * Execute action proxy
     */
    static v1ExecuteActionProxy(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/actions/proxy'
        });
    }
    /**
     * Get action
     */
    static v1GetAction(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/actions/{actionId}'
        });
    }
}
exports.ActionsControllerV1Service = ActionsControllerV1Service;
class ConnectionsService {
    /**
     * Get connections
     */
    static getConnections(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/connectedAccounts'
        });
    }
    /**
     * Initiate connection
     */
    static initiateConnection(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/connectedAccounts'
        });
    }
    /**
     * Get connection
     */
    static getConnection(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/connectedAccounts/{connectedAccountId}'
        });
    }
    /**
     * Delete connection
     */
    static deleteConnection(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).delete({
            ...options,
            url: '/api/v1/connectedAccounts/{connectedAccountId}'
        });
    }
    /**
     * Disable connection
     */
    static disableConnection(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/connectedAccounts/{connectedAccountId}/disable'
        });
    }
    /**
     * Enable connection
     */
    static enableConnection(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/connectedAccounts/{connectedAccountId}/enable'
        });
    }
}
exports.ConnectionsService = ConnectionsService;
class AppLoginService {
    /**
     * Handle o auth 2 callback
     */
    static handleOauth2Callback(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/login/{appName}/auth'
        });
    }
    /**
     * Verify o auth 2 login
     */
    static verifyOauth2Login(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/login/{appName}/callback'
        });
    }
}
exports.AppLoginService = AppLoginService;
class TriggersService {
    /**
     * List triggers
     * List triggers
     */
    static listTriggers(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/triggers'
        });
    }
    /**
     * Handle trigger
     */
    static handleTrigger(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/triggers/handle/{appName}/{clientId}'
        });
    }
    /**
     * Enable trigger
     * Enables a trigger for a connected account and specific trigger name.
     */
    static enableTrigger(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/triggers/enable/{connectedAccountId}/{triggerName}'
        });
    }
    /**
     * Get trigger
     * Retrieves a specific trigger by its ID.
     */
    static getTrigger(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/triggers/get/{triggerId}'
        });
    }
    /**
     * Get active triggers
     * Lists active triggers based on query parameters.
     */
    static getActiveTriggers(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/triggers/active_triggers'
        });
    }
    /**
     * Switch trigger instance status
     * Switches the status of a trigger instance.
     */
    static switchTriggerInstanceStatus(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).patch({
            ...options,
            url: '/api/v1/triggers/instance/{triggerId}/status'
        });
    }
    /**
     * Switch post trigger instance status
     * Switches the status of a trigger instance.
     */
    static switchPostTriggerInstanceStatus(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/triggers/instance/{triggerId}/status'
        });
    }
    /**
     * Disable trigger
     * Disables a specified trigger instance.
     */
    static disableTrigger(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/triggers/disable/{triggerId}'
        });
    }
    /**
     * Delete trigger
     * Deletes a specified trigger instance.
     */
    static deleteTrigger(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).delete({
            ...options,
            url: '/api/v1/triggers/instance/{triggerInstanceId}'
        });
    }
    /**
     * Set callback url
     * Sets a universal callback URL for the client.
     */
    static setCallbackUrl(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/triggers/setCallbackURL'
        });
    }
    /**
     * Set callback url
     * Sets a universal callback URL for the client, ensuring it is updated in the system.
     */
    static setCallbackUrl1(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/triggers/set_callback_url'
        });
    }
    /**
     * Get webhook url
     * Retrieves the universal callback URL set for the client.
     */
    static getWebhookUrl(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/triggers/callback_url'
        });
    }
    /**
     * Get logs
     * Fetches logs based on connection and integration details.
     */
    static getLogs(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/triggers/logs'
        });
    }
    /**
     * Handle pusher events
     */
    static handlePusherEvents(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/triggers/pusher'
        });
    }
}
exports.TriggersService = TriggersService;
class AdminService {
    /**
     * Sentry dns
     */
    static sentryDns(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/cli/sentry-dns'
        });
    }
}
exports.AdminService = AdminService;
class CliService {
    /**
     * Handle cli code exchange
     */
    static handleCliCodeExchange(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/cli/generate-cli-session'
        });
    }
    /**
     * Get cli code
     */
    static getCliCode(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/cli/get-cli-code'
        });
    }
    /**
     * Handle cli code verification
     */
    static handleCliCodeVerification(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/cli/verify-cli-code'
        });
    }
}
exports.CliService = CliService;
class MetadataService {
    /**
     * Get metadata
     */
    static getMetadata(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/metadata'
        });
    }
    /**
     * Update metadata
     */
    static updateMetadata(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/metadata'
        });
    }
    /**
     * Toggle trigger state
     */
    static toggleTriggerState(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v1/metadata/toggle/trigger'
        });
    }
    /**
     * Get toggle info
     */
    static getToggleInfo(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/metadata/toggle.info'
        });
    }
}
exports.MetadataService = MetadataService;
class LogsService {
    /**
     * Get logs
     * List logs
     */
    static getLogs(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v1/logs/'
        });
    }
}
exports.LogsService = LogsService;
class ActionsControllerV2Service {
    /**
     * List actions
     * Retrieve a list of all actions based on query parameters.
     */
    static v2ListActions(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).get({
            ...options,
            url: '/api/v2/actions'
        });
    }
    /**
     * Execute action
     * Execute an action. Support both connected account and no auth auth.
     */
    static v2ExecuteAction(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v2/actions/{actionId}/execute'
        });
    }
    /**
     * Execute action proxy
     * Execute an action with direct auth.
     */
    static v2ExecuteActionProxy(options) {
        var _a;
        return ((_a = options === null || options === void 0 ? void 0 : options.client) !== null && _a !== void 0 ? _a : exports.client).post({
            ...options,
            url: '/api/v2/actions/proxy'
        });
    }
}
exports.ActionsControllerV2Service = ActionsControllerV2Service;
