// composio-core/lib/src/env/docker/workspace.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DockerWorkspace = void 0;
const uuid_1 = require("uuid");
const base_1 = require("../base");
const shared_1 = require("../../utils/shared");
const logger_1 = __importDefault(require("../../utils/logger"));
const ENV_COMPOSIO_DEV_MODE = "COMPOSIO_DEV_MODE";
const ENV_COMPOSIO_SWE_AGENT = "COMPOSIO_SWE_AGENT";
const DEFAULT_IMAGE = "composio/composio";
const DEFAULT_PORT = 54321;
function getFreePort() {
    return new Promise((resolve, reject) => {
        const server = require("node:net").createServer();
        server.unref();
        server.on("error", reject);
        server.listen(0, () => {
            const port = server.address().port;
            server.close(() => resolve(port));
        });
    });
}
class DockerWorkspace extends base_1.RemoteWorkspace {
    constructor(configRepo) {
        super(configRepo);
        this.container = null;
        this.port = DEFAULT_PORT;
        this.url = "";
        this.id = `composio-${(0, uuid_1.v4)()}`;
        this.image = (0, shared_1.getEnvVariable)(ENV_COMPOSIO_SWE_AGENT, DEFAULT_IMAGE);
        this.docker = (0, shared_1.nodeExternalRequire)("dockerode")();
        this._ports = configRepo.config.ports;
        this._volumes = configRepo.config.volumes;
    }
    getBaseDockerConfig() {
        const IS_DEV_MODE = (0, shared_1.getEnvVariable)(ENV_COMPOSIO_DEV_MODE, "0");
        const exposedPorts = {
            "8000/tcp": {},
        };
        const portBindings = {
            "8000/tcp": [
                {
                    HostPort: this.port.toString(),
                },
            ],
        };
        // Add additional ports if specified in the environment configuration
        if (this._ports) {
            for (const port of Object.keys(this._ports)) {
                const portKey = `${port}/tcp`;
                exposedPorts[portKey] = {};
                portBindings[portKey] = [{ HostPort: port }];
            }
        }
        const volumeBindings = [];
        // Add additional volumes if specified in the environment configuration
        if (this._volumes) {
            for (const hostPath in this._volumes) {
                const containerPath = this._volumes[hostPath];
                volumeBindings.push(`${hostPath}:${containerPath}`);
            }
        }
        if (IS_DEV_MODE === "1") {
            const path = require("node:path");
            const os = require("node:os");
            const COMPOSIO_PATH = path.resolve(__dirname, "../../../../python/");
            const COMPOSIO_CACHE = path.join(os.homedir(), ".composio");
            volumeBindings.push(...[
                `${COMPOSIO_PATH}:/opt/composio-core:rw`,
                `${COMPOSIO_CACHE}:/root/.composio:rw`,
            ]);
        }
        const envBindings = Object.entries(this.environment).map(([key, value]) => `${key}=${value}`);
        if (IS_DEV_MODE === "1") {
            envBindings.push(`${ENV_COMPOSIO_DEV_MODE}=1`);
        }
        return { exposedPorts, portBindings, volumeBindings, envBindings };
    }
    async setup() {
        var _a;
        this.port = await getFreePort();
        this.url = `http://localhost:${this.port}/api`;
        const images = await this.docker.listImages();
        const imageExists = images.some((image) => image.RepoTags && image.RepoTags.find((tag) => tag.startsWith(this.image)));
        if (!imageExists) {
            logger_1.default.info(`Pulling Docker image ${this.image}...`);
            let cliProgress = (0, shared_1.nodeExternalRequire)("cli-progress");
            const bar = new cliProgress.SingleBar({
                format: '{bar} | {percentage}% | {status}',
                hideCursor: true
            }, cliProgress.Presets.shades_classic);
            bar.start(100, 0, { status: 'Initializing...' });
            bar.update({ status: `Image ${this.image} not found locally. Pulling from Docker Hub...` });
            await new Promise((resolve, reject) => {
                this.docker.pull(this.image, (err, stream) => {
                    if (err) {
                        bar.stop();
                        logger_1.default.error('Failed to pull Docker image.');
                        return reject(err);
                    }
                    this.docker.modem.followProgress(stream, onFinished, onProgress);
                    function onFinished(err, output) {
                        if (err) {
                            bar.stop();
                            logger_1.default.error('Failed to pull Docker image.');
                            return reject(err);
                        }
                        bar.update(100, { status: 'Docker image pulled successfully.' });
                        bar.stop();
                        resolve(output);
                    }
                    function onProgress(event) {
                        bar.update({ status: event.status });
                    }
                });
            });
        }
        else {
            logger_1.default.debug(`Image ${this.image} found locally.`);
        }
        const containers = await this.docker.listContainers({ all: true });
        const existingContainer = containers.find((container) => container.Names.find((name) => name.startsWith(`/composio-`)));
        if (existingContainer) {
            logger_1.default.debug(`Container with name ${this.id} is already running.`);
            this.container = this.docker.getContainer(existingContainer.Id);
            await this.container.restart();
            this.port = (_a = existingContainer.Ports.find((port) => port.PrivatePort === 8000)) === null || _a === void 0 ? void 0 : _a.PublicPort;
            this.url = `http://localhost:${this.port}/api`;
        }
        else {
            const { exposedPorts, portBindings, volumeBindings, envBindings } = this.getBaseDockerConfig();
            const containerOptions = {
                Image: this.image,
                name: this.id,
                Cmd: ["/root/entrypoint.sh"],
                Tty: true,
                AttachStdin: true,
                AttachStdout: true,
                AttachStderr: true,
                OpenStdin: true,
                StdinOnce: false,
                ExposedPorts: exposedPorts,
                HostConfig: {
                    PortBindings: portBindings,
                    Binds: volumeBindings,
                },
                Env: envBindings,
            };
            this.container = await this.docker.createContainer(containerOptions);
            await this.container.start();
        }
        await this.waitForContainer();
    }
    async waitForContainer() {
        while (true) {
            try {
                const response = await fetch(this.url, {
                    method: "GET",
                    headers: {
                        "Content-Type": "application/json",
                        "x-api-key": this.environment.composioAPIKey,
                    },
                });
                if (response.ok) {
                    return;
                }
            }
            catch (error) {
                await new Promise((resolve) => setTimeout(resolve, 100));
            }
        }
    }
    async teardown() {
        if (this.container) {
            logger_1.default.info(`Stopping container ${this.container.id}...`);
            try {
                await this.container.kill();
                await this.container.remove();
            }
            catch (error) {
                logger_1.default.debug("Failed to stop and remove container:", error);
            }
        }
    }
}
exports.DockerWorkspace = DockerWorkspace;
