// composio-core/lib/src/sdk/models/actions.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Actions = void 0;
const client_1 = __importDefault(require("../client/client"));
class Actions {
    constructor(backendClient) {
        this.backendClient = backendClient;
    }
    /**
     * Retrieves details of a specific action in the Composio platform by providing its action name.
     *
     * The response includes the action's name, display name, description, input parameters, expected response, associated app information, and enabled status.
     *
     * @param {GetActionData} data The data for the request.
     * @returns {CancelablePromise<GetActionResponse[0]>} A promise that resolves to the details of the action.
     * @throws {ApiError} If the request fails.
     */
    async get(data) {
        const actions = await client_1.default.actionsV1.v1GetAction({
            path: {
                actionId: data.actionName
            }
        });
        return actions.data[0];
    }
    /**
     * Retrieves a list of all actions in the Composio platform.
     *
     * This method allows you to fetch a list of all the available actions. It supports pagination to handle large numbers of actions. The response includes an array of action objects, each containing information such as the action's name, display name, description, input parameters, expected response, associated app information, and enabled status.
     *
     * @param {GetListActionsData} data The data for the request.
     * @returns {CancelablePromise<GetListActionsResponse>} A promise that resolves to the list of all actions.
     * @throws {ApiError} If the request fails.
     */
    list(data = {}) {
        return client_1.default.actionsV2.v2ListActions({
            query: {
                actions: data.actions,
                apps: data.apps,
                showAll: data.showAll,
                tags: data.tags,
                useCase: data.useCase,
                filterImportantActions: data.filterImportantActions,
                showEnabledOnly: data.showEnabledOnly
            }
        }).then(res => {
            return res.data;
        });
    }
    /**
     * Executes a specific action in the Composio platform.
     *
     * This method allows you to trigger the execution of an action by providing its name and the necessary input parameters. The request includes the connected account ID to identify the app connection to use for the action, and the input parameters required by the action. The response provides details about the execution status and the response data returned by the action.
     *
     * @param {ExecuteActionData} data The data for the request.
     * @returns {CancelablePromise<ExecuteActionResponse>} A promise that resolves to the execution status and response data.
     * @throws {ApiError} If the request fails.
     */
    async execute(data) {
        const { data: res, error } = await client_1.default.actionsV2.v2ExecuteAction({
            body: data.requestBody,
            path: {
                actionId: data.actionName
            }
        });
        return res;
    }
}
exports.Actions = Actions;
