// composio-core/lib/src/sdk/models/actions.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const globals_1 = require("@jest/globals");
const getBackendClient_1 = require("../testUtils/getBackendClient");
const actions_1 = require("./actions");
const connectedAccounts_1 = require("./connectedAccounts");
(0, globals_1.describe)("Apps class tests", () => {
    let backendClient;
    let actions;
    let connectedAccouns;
    (0, globals_1.beforeAll)(() => {
        backendClient = (0, getBackendClient_1.getBackendClient)();
        actions = new actions_1.Actions(backendClient);
        connectedAccouns = new connectedAccounts_1.ConnectedAccounts(backendClient);
    });
    (0, globals_1.it)("should get a specific action", async () => {
        const actionName = "GITHUB_GITHUB_API_ROOT";
        const action = await actions.get({ actionName: actionName.toLowerCase() });
        (0, globals_1.expect)(action).toHaveProperty('name', actionName);
    });
    (0, globals_1.it)("should get a list of actions", async () => {
        const actionsList = await actions.list();
        (0, globals_1.expect)(actionsList.items).toBeInstanceOf(Array);
        (0, globals_1.expect)(actionsList.items).not.toHaveLength(0);
    });
    (0, globals_1.it)("should execute an action with a connected account for GitHub", async () => {
        const actionName = "GITHUB_GITHUB_API_ROOT".toLowerCase();
        const connectedAccountsResult = await connectedAccouns.list({ appNames: 'github', status: 'ACTIVE' });
        (0, globals_1.expect)(connectedAccountsResult.items).not.toHaveLength(0);
        const connectionId = connectedAccountsResult.items[0].id;
        const executionResult = await actions.execute({
            actionName: actionName,
            requestBody: {
                connectedAccountId: connectionId,
                input: {},
                appName: 'github'
            }
        });
        (0, globals_1.expect)(executionResult).toHaveProperty('successfull', true);
        (0, globals_1.expect)(executionResult.data).toHaveProperty('authorizations_url');
    });
    (0, globals_1.it)("should execute an action of noauth app", async () => {
        const actionName = "codeinterpreter_execute_code";
        const input = { code_to_execute: 'print("Hello World");' };
        const executionResult = await actions.execute({
            actionName,
            requestBody: {
                input: input,
                appName: "codeinterpreter",
            }
        });
        (0, globals_1.expect)(executionResult).toHaveProperty('successfull', true);
        //@ts-ignore
        (0, globals_1.expect)(executionResult.data).toHaveProperty('stdout', 'Hello World\n');
        (0, globals_1.expect)(executionResult.data).toHaveProperty('stderr', '');
    });
});
