// composio-core/lib/src/sdk/models/backendClient.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BackendClient = void 0;
const client_1 = __importDefault(require("../client/client"));
const services_gen_1 = require("../client/services.gen");
/**
 * Class representing the details required to initialize and configure the API client.
 */
class BackendClient {
    /**
     * Creates an instance of apiClientDetails.
     * @param {string} apiKey - The API key for client initialization.
     * @param {string} baseUrl - The base URL for the API client.
     * @param {string} runtime - The runtime environment identifier.
     * @throws Will throw an error if the API key is not provided.
     */
    constructor(apiKey, baseUrl, runtime) {
        this.runtime = runtime || '';
        this.apiKey = apiKey;
        this.baseUrl = baseUrl;
        if (!apiKey) {
            throw new Error(`API Key is required for initializing the client`);
        }
        // Validate baseUrl
        if (!baseUrl.startsWith("http://") && !baseUrl.startsWith("https://")) {
            throw new Error(`Base URL is not valid, got ${baseUrl}`);
        }
        this.initializeApiClient();
    }
    /**
     * Retrieves the client ID from the user's information.
     * @returns {Promise<string>} A promise that resolves to the client ID.
     * @throws Will throw an error if the HTTP request fails.
     */
    async getClientId() {
        const response = await client_1.default.clientAuthService.getUserInfo();
        if (response.status !== 200) {
            throw new Error(`HTTP Error: ${response.status}`);
        }
        return response.data.client.id;
    }
    /**
     * Initializes the API client with the provided configuration.
     * @private
     */
    initializeApiClient() {
        services_gen_1.client.setConfig({
            baseURL: this.baseUrl,
            headers: {
                'X-API-KEY': `${this.apiKey}`,
                'X-SOURCE': 'js_sdk',
                'X-RUNTIME': this.runtime
            }
        });
    }
}
exports.BackendClient = BackendClient;
