// composio-core/lib/src/sdk/models/connectedAccounts.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionRequest = exports.ConnectedAccounts = void 0;
const client_1 = __importDefault(require("../client/client"));
const client_2 = __importDefault(require("../client/client"));
class ConnectedAccounts {
    constructor(backendClient) {
        this.backendClient = backendClient;
    }
    list(data = {}) {
        return client_2.default.connections.getConnections({
            query: data
        }).then(res => {
            return res.data;
        });
    }
    create(data = {}) {
        return client_2.default.connections.initiateConnection({
            body: data
        }).then(res => res.data);
    }
    get(data) {
        return client_2.default.connections.getConnection({
            path: data
        }).then(res => res.data);
    }
    async initiate(data) {
        const res = await client_1.default.connections.initiateConnection({ body: data }).then(res => res.data);
        //@ts-ignore
        return new ConnectionRequest(res === null || res === void 0 ? void 0 : res.connectionStatus, res === null || res === void 0 ? void 0 : res.connectedAccountId, res === null || res === void 0 ? void 0 : res.redirectUrl);
    }
}
exports.ConnectedAccounts = ConnectedAccounts;
class ConnectionRequest {
    /**
     * Connection request model.
     * @param {string} connectionStatus The status of the connection.
     * @param {string} connectedAccountId The unique identifier of the connected account.
     * @param {string} [redirectUrl] The redirect URL for completing the connection flow.
     */
    constructor(connectionStatus, connectedAccountId, redirectUrl = null) {
        this.connectionStatus = connectionStatus;
        this.connectedAccountId = connectedAccountId;
        this.redirectUrl = redirectUrl;
    }
    /**
     * Save user access data.
     * @param {Composio} client The Composio client instance.
     * @param {Object} data The data to save.
     * @param {Object} data.fieldInputs The field inputs to save.
     * @param {string} [data.redirectUrl] The redirect URL for completing the connection flow.
     * @param {string} [data.entityId] The entity ID associated with the user.
     * @returns {Promise<Object>} The response from the server.
     */
    async saveUserAccessData(data) {
        const connectedAccount = await client_2.default.connections.getConnection({
            path: {
                connectedAccountId: this.connectedAccountId
            }
        });
        return client_2.default.connections.initiateConnection({
            body: {
                // @ts-ignore
                integrationId: connectedAccount.integrationId,
                //@ts-ignore
                data: data.fieldInputs,
                redirectUri: data.redirectUrl,
                userUuid: data.entityId,
            }
        });
    }
    /**
     * Wait until the connection becomes active.
     * @param {Composio} client The Composio client instance.
     * @param {number} [timeout=60] The timeout period in seconds.
     * @returns {Promise<ConnectedAccountModel>} The connected account model.
     * @throws {ComposioClientError} If the connection does not become active within the timeout period.
     */
    async waitUntilActive(timeout = 60) {
        const startTime = Date.now();
        while (Date.now() - startTime < timeout * 1000) {
            // @ts-ignore
            const connection = await client_2.default.connections.getConnection({
                path: {
                    connectedAccountId: this.connectedAccountId
                }
            }).then(res => res.data);
            //@ts-ignore
            if (connection.status === 'ACTIVE') {
                return connection;
            }
            await new Promise(resolve => setTimeout(resolve, 1000));
        }
        throw new Error('Connection did not become active within the timeout period.');
    }
}
exports.ConnectionRequest = ConnectionRequest;
