// composio-core/lib/src/sdk/models/Entity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Entity = void 0;
const actions_1 = require("./actions");
const apps_1 = require("./apps");
const integrations_1 = require("./integrations");
const activeTriggers_1 = require("./activeTriggers");
const connectedAccounts_1 = require("./connectedAccounts");
const triggers_1 = require("./triggers");
class Entity {
    constructor(backendClient, id = 'default') {
        this.backendClient = backendClient;
        this.id = id;
        this.triggerModel = new triggers_1.Triggers(this.backendClient);
        this.actionsModel = new actions_1.Actions(this.backendClient);
        this.apps = new apps_1.Apps(this.backendClient);
        this.connectedAccounts = new connectedAccounts_1.ConnectedAccounts(this.backendClient);
        this.integrations = new integrations_1.Integrations(this.backendClient);
        this.activeTriggers = new activeTriggers_1.ActiveTriggers(this.backendClient);
    }
    async execute(actionName, params, text, connectedAccountId) {
        const action = await this.actionsModel.get({
            actionName: actionName
        });
        if (!action) {
            throw new Error("Could not find action: " + actionName);
        }
        const app = await this.apps.get({
            appKey: action.appKey
        });
        if (app.yaml.no_auth) {
            return this.actionsModel.execute({
                actionName: actionName,
                requestBody: {
                    input: params,
                    appName: action.appKey
                }
            });
        }
        let connectedAccount = null;
        if (connectedAccountId) {
            connectedAccount = await this.connectedAccounts.get({
                connectedAccountId: connectedAccountId
            });
        }
        else {
            const connectedAccounts = await this.connectedAccounts.list({
                user_uuid: this.id,
                appNames: action.appKey,
                status: 'ACTIVE'
            });
            // @ts-ignore
            if ((connectedAccounts === null || connectedAccounts === void 0 ? void 0 : connectedAccounts.items.length) === 0) {
                throw new Error('No connected account found');
            }
            // @ts-ignore
            connectedAccount = connectedAccounts.items[0];
        }
        return this.actionsModel.execute({
            actionName: actionName,
            requestBody: {
                // @ts-ignore
                connectedAccountId: connectedAccount === null || connectedAccount === void 0 ? void 0 : connectedAccount.id,
                input: params,
                appName: action.appKey,
                text: text
            }
        });
    }
    async getConnection(app, connectedAccountId) {
        if (connectedAccountId) {
            return await this.connectedAccounts.get({
                connectedAccountId
            });
        }
        let latestAccount = null;
        let latestCreationDate = null;
        const connectedAccounts = await this.connectedAccounts.list({
            user_uuid: this.id,
        });
        if (!connectedAccounts.items || connectedAccounts.items.length === 0) {
            return null;
        }
        for (const connectedAccount of connectedAccounts.items) {
            if (app === connectedAccount.appName) {
                const creationDate = new Date(connectedAccount.createdAt);
                if ((!latestAccount || (latestCreationDate && creationDate > latestCreationDate)) && connectedAccount.status === "ACTIVE") {
                    latestCreationDate = creationDate;
                    latestAccount = connectedAccount;
                }
            }
        }
        if (!latestAccount) {
            return null;
        }
        return this.connectedAccounts.get({
            connectedAccountId: latestAccount.id
        });
    }
    async setupTrigger(app, triggerName, config) {
        const connectedAccount = await this.getConnection(app);
        if (!connectedAccount) {
            throw new Error(`Could not find a connection with app='${app}' and entity='${this.id}'`);
        }
        const trigger = await this.triggerModel.setup(connectedAccount.id, triggerName, config);
        return trigger;
    }
    async disableTrigger(triggerId) {
        return activeTriggers_1.ActiveTriggers.disable({ triggerId: triggerId });
    }
    async getConnections() {
        /**
         * Get all connections for an entity.
         */
        const connectedAccounts = await this.connectedAccounts.list({
            user_uuid: this.id
        });
        return connectedAccounts.items;
    }
    async getActiveTriggers() {
        /**
         * Get all active triggers for an entity.
         */
        const connectedAccounts = await this.getConnections();
        const activeTriggers = await this.activeTriggers.list({
            connectedAccountIds: connectedAccounts.map((account) => account.id).join(",")
        });
        return activeTriggers;
    }
    async initiateConnection(appName, authMode, authConfig, redirectUrl, integrationId) {
        // Get the app details from the client
        const app = await this.apps.get({ appKey: appName });
        const timestamp = new Date().toISOString().replace(/[-:.]/g, "");
        let integration = integrationId ? await this.integrations.get({ integrationId: integrationId }) : null;
        // Create a new integration if not provided
        if (!integration && authMode) {
            integration = await this.integrations.create({
                appId: app.appId,
                name: `integration_${timestamp}`,
                authScheme: authMode,
                authConfig: authConfig,
                useComposioAuth: false,
            });
        }
        if (!integration && !authMode) {
            integration = await this.integrations.create({
                appId: app.appId,
                name: `integration_${timestamp}`,
                useComposioAuth: true,
            });
        }
        // Initiate the connection process
        return this.connectedAccounts.initiate({
            integrationId: integration.id,
            userUuid: this.id,
            redirectUri: redirectUrl,
        });
    }
}
exports.Entity = Entity;
