// composio-core/lib/src/sdk/models/integrations.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const globals_1 = require("@jest/globals");
const getBackendClient_1 = require("../testUtils/getBackendClient");
const integrations_1 = require("./integrations");
(0, globals_1.describe)("Integrations class tests", () => {
    let backendClient;
    let integrations;
    let createdIntegrationId;
    (0, globals_1.beforeAll)(() => {
        backendClient = (0, getBackendClient_1.getBackendClient)();
        integrations = new integrations_1.Integrations(backendClient);
    });
    (0, globals_1.it)("Retrieve integrations list", async () => {
        const integrationsList = await integrations.list();
        (0, globals_1.expect)(integrationsList === null || integrationsList === void 0 ? void 0 : integrationsList.items).toBeInstanceOf(Array);
        (0, globals_1.expect)(integrationsList === null || integrationsList === void 0 ? void 0 : integrationsList.items).not.toHaveLength(0);
    });
    (0, globals_1.it)("should create an integration and verify its properties", async () => {
        const integrationCreation = await integrations.create({
            appId: "01e22f33-dc3f-46ae-b58d-050e4d2d1909",
            name: "test_integration_220",
            useComposioAuth: true,
            // @ts-ignore
            forceNewIntegration: true
        });
        (0, globals_1.expect)(integrationCreation.id).toBeTruthy();
        (0, globals_1.expect)(integrationCreation.appName).toBe("github");
        createdIntegrationId = integrationCreation.id;
    });
    (0, globals_1.it)("should retrieve the created integration by ID and verify its properties", async () => {
        const integration = await integrations.get({
            integrationId: createdIntegrationId
        });
        (0, globals_1.expect)(integration.id).toBe(createdIntegrationId);
        (0, globals_1.expect)(integration.appId).toBe("01e22f33-dc3f-46ae-b58d-050e4d2d1909");
        (0, globals_1.expect)(integration.authScheme).toBe("OAUTH2");
    });
    (0, globals_1.it)("should delete the created integration", async () => {
        if (!createdIntegrationId)
            return;
        await integrations.delete({
            path: {
                integrationId: createdIntegrationId
            }
        });
    });
});
