// composio-core/lib/src/sdk/models/integrations.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Integrations = void 0;
const client_1 = __importDefault(require("../client/client"));
class Integrations {
    constructor(backendClient) {
        this.backendClient = backendClient;
    }
    /**
     * Retrieves a list of all available integrations in the Composio platform.
     *
     * This method allows clients to explore and discover the supported integrations. It returns an array of integration objects, each containing essential details such as the integration's key, name, description, logo, categories, and unique identifier.
     *
     * @returns {Promise<ListAllIntegrationsResponse>} A promise that resolves to the list of all integrations.
     * @throws {ApiError} If the request fails.
     */
    list(data = {}) {
        return client_1.default.appConnector.listGlobalConnectors({
            query: data
        }).then(res => res.data);
    }
    /**
     * Retrieves details of a specific integration in the Composio platform by providing its integration name.
     *
     * The response includes the integration's name, display name, description, input parameters, expected response, associated app information, and enabled status.
     *
     * @param {GetIntegrationData} data The data for the request.
     * @returns {CancelablePromise<GetIntegrationResponse>} A promise that resolves to the details of the integration.
     * @throws {ApiError} If the request fails.
     */
    get(data) {
        return client_1.default.appConnector.getConnectorInfo({
            path: data
        }).then(res => {
            return res.data;
        });
    }
    /**
     * Creates a new integration in the Composio platform.
     *
     * This method allows clients to create a new integration by providing the necessary details such as app ID, name, authentication mode, and configuration.
     *
     * @param {CreateIntegrationData["requestBody"]} data The data for the request.
     * @returns {CancelablePromise<CreateIntegrationResponse>} A promise that resolves to the created integration model.
     * @throws {ApiError} If the request fails.
     */
    create(data) {
        if (!(data === null || data === void 0 ? void 0 : data.authConfig)) {
            data.authConfig = {};
        }
        return client_1.default.appConnector.createConnector({
            body: {
                name: data === null || data === void 0 ? void 0 : data.name,
                appId: data === null || data === void 0 ? void 0 : data.appId,
                authConfig: data === null || data === void 0 ? void 0 : data.authConfig,
                authScheme: data === null || data === void 0 ? void 0 : data.authScheme,
                useComposioAuth: data === null || data === void 0 ? void 0 : data.useComposioAuth,
                forceNewIntegration: true
            }
        }).then(res => {
            return res.data;
        });
    }
    delete(data) {
        return client_1.default.appConnector.deleteConnector(data).then(res => res.data);
    }
}
exports.Integrations = Integrations;
