// composio-core/lib/src/sdk/models/triggers.spec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const globals_1 = require("@jest/globals");
const getBackendClient_1 = require("../testUtils/getBackendClient");
const triggers_1 = require("./triggers");
const connectedAccounts_1 = require("./connectedAccounts");
const Entity_1 = require("./Entity");
const actions_1 = require("./actions");
(0, globals_1.describe)("Apps class tests", () => {
    let backendClient;
    let triggers;
    let connectedAccounts;
    let entity;
    let triggerId;
    let actions;
    (0, globals_1.beforeAll)(() => {
        backendClient = (0, getBackendClient_1.getBackendClient)();
        triggers = new triggers_1.Triggers(backendClient);
        connectedAccounts = new connectedAccounts_1.ConnectedAccounts(backendClient);
        entity = new Entity_1.Entity(backendClient, "default");
        connectedAccounts = new connectedAccounts_1.ConnectedAccounts(backendClient);
        actions = new actions_1.Actions(backendClient);
    });
    (0, globals_1.it)("should create an Apps instance and retrieve apps list", async () => {
        const triggerList = await triggers.list();
        (0, globals_1.expect)(triggerList.length).toBeGreaterThan(0);
    });
    (0, globals_1.it)("should retrieve a list of triggers for a specific app", async () => {
        const triggerList = await triggers.list({
            appNames: "github"
        });
        // this is breaking for now
        (0, globals_1.expect)(triggerList.length).toBeGreaterThan(0);
        (0, globals_1.expect)(triggerList[0].appName).toBe("github");
    });
});
(0, globals_1.describe)("Apps class tests subscribe", () => {
    let backendClient;
    let triggers;
    let connectedAccounts;
    let actions;
    let entity;
    let triggerId;
    (0, globals_1.beforeAll)(() => {
        backendClient = (0, getBackendClient_1.getBackendClient)();
        triggers = new triggers_1.Triggers(backendClient);
        connectedAccounts = new connectedAccounts_1.ConnectedAccounts(backendClient);
        entity = new Entity_1.Entity(backendClient, "default");
        actions = new actions_1.Actions(backendClient);
    });
    (0, globals_1.it)("should create a new trigger for gmail", async () => {
        var _a;
        const connectedAccount = await connectedAccounts.list({ query: { user_uuid: 'default' } });
        const connectedAccountId = (_a = connectedAccount.items.find(item => item.appName === 'gmail')) === null || _a === void 0 ? void 0 : _a.id;
        const trigger = await triggers.setup(connectedAccountId, 'gmail_new_gmail_message', {
            "userId": connectedAccount.items[0].id,
            "interval": 60,
            "labelIds": "INBOX",
        });
        (0, globals_1.expect)(trigger.status).toBe("success");
        (0, globals_1.expect)(trigger.triggerId).toBeTruthy();
        triggerId = trigger.triggerId;
    });
    (0, globals_1.it)("should disable, enable, and then disable the created trigger", async () => {
        let trigger = await triggers.disable({ triggerId });
        (0, globals_1.expect)(trigger.status).toBe("success");
        trigger = await triggers.enable({ triggerId });
        (0, globals_1.expect)(trigger.status).toBe("success");
        trigger = await triggers.disable({ triggerId });
        (0, globals_1.expect)(trigger.status).toBe("success");
    });
    (0, globals_1.it)("should subscribe to a trigger and receive a trigger", async () => {
        function waitForTriggerReceived() {
            return new Promise((resolve) => {
                triggers.subscribe((data) => {
                    resolve(data);
                }, {
                    appName: "github",
                    triggerName: "GITHUB_ISSUE_ADDED_EVENT"
                });
                setTimeout(async () => {
                    const actionName = "github_create_an_issue";
                    // Not urgent
                    const connectedAccountsResult = await connectedAccounts.list({ integrationId: 'ca85b86b-1198-4e1a-8d84-b14640564c77' });
                    const connectionId = connectedAccountsResult.items[0].id;
                    await actions.execute({
                        actionName,
                        requestBody: {
                            connectedAccountId: connectionId,
                            input: {
                                title: "test",
                                owner: "ComposioHQ",
                                repo: "test_repo",
                            },
                            appName: 'github'
                        }
                    });
                }, 4000);
            });
        }
        const data = await waitForTriggerReceived();
        //@ts-ignore
        (0, globals_1.expect)(data.payload.triggerName).toBe("GITHUB_ISSUE_ADDED_EVENT");
        triggers.unsubscribe();
    });
});
