// composio-core/lib/src/sdk/models/triggers.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Triggers = void 0;
const pusher_1 = require("../utils/pusher");
const logger_1 = __importDefault(require("../../utils/logger"));
const client_1 = __importDefault(require("../client/client"));
class Triggers {
    constructor(backendClient) {
        this.trigger_to_client_event = "trigger_to_client";
        this.backendClient = backendClient;
    }
    /**
     * Retrieves a list of all triggers in the Composio platform.
     *
     * This method allows you to fetch a list of all the available triggers. It supports pagination to handle large numbers of triggers. The response includes an array of trigger objects, each containing information such as the trigger's name, description, input parameters, expected response, associated app information, and enabled status.
     *
     * @param {ListTriggersData} data The data for the request.
     * @returns {CancelablePromise<ListTriggersResponse>} A promise that resolves to the list of all triggers.
     * @throws {ApiError} If the request fails.
     */
    //@ts-ignore
    list(data = {}) {
        //@ts-ignore
        return client_1.default.triggers.listTriggers({
            query: {
                appNames: data === null || data === void 0 ? void 0 : data.appNames,
            }
        }).then(res => res.data);
    }
    /**
     * Setup a trigger for a connected account.
     *
     * @param {SetupTriggerData} data The data for the request.
     * @returns {CancelablePromise<SetupTriggerResponse>} A promise that resolves to the setup trigger response.
     * @throws {ApiError} If the request fails.
     */
    //@ts-ignore
    async setup(connectedAccountId, triggerName, config) {
        //@ts-ignore
        const { data, error } = await client_1.default.triggers.enableTrigger({
            path: {
                connectedAccountId,
                triggerName
            },
            body: {
                triggerConfig: config
            }
        });
        if (error) {
            console.log(error);
            throw Error("Failed to setup trigger");
        }
        return data;
    }
    enable(data) {
        return client_1.default.triggers.switchTriggerInstanceStatus({
            path: data,
            body: {
                enabled: true
            }
        }).then(res => res.data);
    }
    disable(data) {
        return client_1.default.triggers.switchTriggerInstanceStatus({
            path: data,
            body: {
                enabled: false
            }
        }).then(res => res.data);
    }
    async subscribe(fn, filters = {}) {
        if (!fn)
            throw new Error("Function is required for trigger subscription");
        //@ts-ignore
        const clientId = await this.backendClient.getClientId();
        //@ts-ignore
        await pusher_1.PusherUtils.getPusherClient(this.backendClient.baseUrl, this.backendClient.apiKey);
        const shouldSendTrigger = (data) => {
            if (Object.keys(filters).length === 0)
                return true;
            else {
                return ((!filters.appName || data.appName === filters.appName) &&
                    (!filters.triggerId || data.metadata.id === filters.triggerId) &&
                    (!filters.connectionId || data.metadata.connectionId === filters.connectionId) &&
                    (!filters.triggerName || data.metadata.triggerName === filters.triggerName) &&
                    (!filters.entityId || data.metadata.connection.clientUniqueUserId === filters.entityId) &&
                    (!filters.integrationId || data.metadata.connection.integrationId === filters.integrationId));
            }
        };
        logger_1.default.info("Subscribing to triggers", filters);
        pusher_1.PusherUtils.triggerSubscribe(clientId, (data) => {
            if (shouldSendTrigger(data)) {
                fn(data);
            }
        });
    }
    async unsubscribe() {
        //@ts-ignore
        const clientId = await this.backendClient.getClientId();
        pusher_1.PusherUtils.triggerUnsubscribe(clientId);
    }
}
exports.Triggers = Triggers;
