// composio-core/lib/src/sdk/utils/config.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.setCliConfig = exports.getAPISDK = exports.getClientBaseConfig = exports.getUserDataJson = exports.userDataPath = void 0;
const fs = __importStar(require("fs"));
const path = __importStar(require("path"));
const os = __importStar(require("os"));
const shared_1 = require("../../utils/shared");
const services_gen_1 = require("../client/services.gen");
const client_1 = __importDefault(require("../client/client"));
// Constants
const LOCAL_CACHE_DIRECTORY_NAME = '.composio';
const USER_DATA_FILE_NAME = 'user_data.json';
const DEFAULT_BASE_URL = "https://backend.composio.dev";
exports.userDataPath = path.join(os.homedir(), LOCAL_CACHE_DIRECTORY_NAME, USER_DATA_FILE_NAME);
const getUserDataJson = () => {
    try {
        const data = fs.readFileSync(exports.userDataPath, 'utf8');
        return JSON.parse(data);
    }
    catch (error) {
        return (error.code === 'ENOENT') ? {} : {};
    }
};
exports.getUserDataJson = getUserDataJson;
// Client configuration
function getClientBaseConfig(baseUrl, apiKey) {
    const userData = (0, exports.getUserDataJson)();
    const { api_key: apiKeyFromUserConfig, base_url: baseURLFromUserConfig } = userData;
    const baseURLParsed = baseUrl || (0, shared_1.getEnvVariable)("COMPOSIO_BASE_URL") || baseURLFromUserConfig || DEFAULT_BASE_URL;
    const apiKeyParsed = apiKey || (0, shared_1.getEnvVariable)("COMPOSIO_API_KEY") || apiKeyFromUserConfig || '';
    return { baseURL: baseURLParsed, apiKey: apiKeyParsed };
}
exports.getClientBaseConfig = getClientBaseConfig;
function getAPISDK(baseUrl, apiKey) {
    const { baseURL, apiKey: apiKeyParsed } = getClientBaseConfig(baseUrl, apiKey);
    services_gen_1.client.setConfig({
        baseURL: baseURL,
        headers: {
            'X-API-KEY': `${apiKeyParsed}`,
            'X-SOURCE': 'js_sdk',
            'X-RUNTIME': 'js_sdk'
        },
        throwOnError: true
    });
    return client_1.default;
}
exports.getAPISDK = getAPISDK;
function setCliConfig(apiKey, baseUrl) {
    const userData = (0, exports.getUserDataJson)();
    userData.api_key = apiKey;
    if (!!baseUrl) {
        userData.base_url = baseUrl;
    }
    fs.writeFileSync(exports.userDataPath, JSON.stringify(userData, null, 2));
}
exports.setCliConfig = setCliConfig;
