// composio-core/lib/src/sdk/utils/pusher.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.PusherUtils = void 0;
const logger_1 = __importDefault(require("../../utils/logger"));
const PUSHER_KEY = process.env.CLIENT_PUSHER_KEY || "ff9f18c208855d77a152";
const PUSHER_CLUSTER = "mt1";
class PusherUtils {
    static getPusherClient(baseURL, apiKey) {
        if (!PusherUtils.pusherClient) {
            const PusherClient = require("pusher-js");
            PusherUtils.pusherClient = new PusherClient(PUSHER_KEY, {
                cluster: PUSHER_CLUSTER,
                channelAuthorization: {
                    endpoint: `${baseURL}/api/v1/client/auth/pusher_auth`,
                    headers: {
                        "x-api-key": apiKey
                    },
                    transport: "ajax",
                }
            });
        }
        return PusherUtils.pusherClient;
    }
    /**
     * Subscribes to a Pusher channel and binds an event to a callback function.
     * @param {string} channelName - The name of the channel to subscribe to.
     * @param {string} event - The event to bind to the channel.
     * @param {(data: any) => void} fn - The callback function to execute when the event is triggered.
     * @returns {PusherClient} The Pusher client instance.
     */
    static async subscribe(channelName, event, fn) {
        try {
            await PusherUtils.pusherClient.subscribe(channelName).bind(event, fn);
        }
        catch (error) {
            logger_1.default.error(`Error subscribing to ${channelName} with event ${event}: ${error}`);
        }
    }
    /**
     * Unsubscribes from a Pusher channel.
     * @param {string} channelName - The name of the channel to unsubscribe from.
     * @returns {void}
     */
    static async unsubscribe(channelName) {
        PusherUtils.pusherClient.unsubscribe(channelName);
    }
    /**
     * Binds an event to a channel with support for chunked messages.
     * @param {PusherClient} channel - The Pusher channel to bind the event to.
     * @param {string} event - The event to bind to the channel.
     * @param {(data: any) => void} callback - The callback function to execute when the event is triggered.
     */
    static bindWithChunking(channel, event, callback) {
        channel.bind(event, callback); // Allow normal unchunked events.
        // Now the chunked variation. Allows arbitrarily long messages.
        const events = {};
        channel.bind("chunked-" + event, (data) => {
            if (!events.hasOwnProperty(data.id)) {
                events[data.id] = { chunks: [], receivedFinal: false };
            }
            const ev = events[data.id];
            ev.chunks[data.index] = data.chunk;
            if (data.final)
                ev.receivedFinal = true;
            if (ev.receivedFinal && ev.chunks.length === Object.keys(ev.chunks).length) {
                callback(JSON.parse(ev.chunks.join("")));
                delete events[data.id];
            }
        });
    }
    /**
     * Subscribes to a trigger channel for a client and handles chunked data.
     * @param {string} clientId - The unique identifier for the client subscribing to the events.
     * @param {(data: any) => void} fn - The callback function to execute when trigger data is received.
     */
    static triggerSubscribe(clientId, fn) {
        var channel = PusherUtils.pusherClient.subscribe(`private-${clientId}_triggers`);
        PusherUtils.bindWithChunking(channel, "trigger_to_client", fn);
        logger_1.default.info(`Subscribed to ${clientId}_triggers`);
    }
    static triggerUnsubscribe(clientId) {
        PusherUtils.pusherClient.unsubscribe(`${clientId}_triggers`);
    }
}
exports.PusherUtils = PusherUtils;
